<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!-- saved from url=(0064)https://www.r-bloggers.com/genome-wide-association-studies-in-r/ -->
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" prefix="og: http://ogp.me/ns#" itemscope="" itemtype="http://schema.org/WebSite"><head profile="http://gmpg.org/xfn/11"><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">

<title>Genome-wide association studies in R | R-bloggers</title>

<script async="" src="./Genome-wide association studies in R _ R-bloggers_files/async-ads.js.download"></script><script src="./Genome-wide association studies in R _ R-bloggers_files/cb=gapi.loaded_0" async=""></script><script type="text/javascript" async="" src="./Genome-wide association studies in R _ R-bloggers_files/cse.js.download"></script><script async="" src="./Genome-wide association studies in R _ R-bloggers_files/analytics.js.download"></script><script type="text/javascript" async="" src="./Genome-wide association studies in R _ R-bloggers_files/vglnk.js.download"></script><script async="" src="./Genome-wide association studies in R _ R-bloggers_files/analytics.js.download"></script><script src="./Genome-wide association studies in R _ R-bloggers_files/pAx-dKbLyle7umPnDyx7D0pQbKA.js.download"></script><script src="./Genome-wide association studies in R _ R-bloggers_files/XwzxUUnPwVYUc-cNUnfA25jqByU.js.download"></script><link rel="stylesheet" href="./Genome-wide association studies in R _ R-bloggers_files/style.css" type="text/css" media="screen">

<style type="text/css">

body { width: 1024px; }

#mainwrapper { width: 1004px; }

#sidebar { width: 180px; }

#sidebar .side-widget { width: 160px; }


#secondsidebar { width: 300px; }

#secondsidebar .side-widget { width: 300px; }


#leftcontent, #twocol, #threecol, #threecol2, .commentlist { width: 456px; }

#leftcontent img, .wp-caption { max-width: 456px; }

#leftcontent .wp-caption img, #leftcontent .gallery-thumb img { max-width: 444px; }

<!-- fix for IE & Safari -->

.threepost embed { width: 132.24px !important; }

.twopost embed { width: 209.76px !important; }

<!-- end fix for Safari -->

</style>

	<link rel="pingback" href="https://www.r-bloggers.com/xmlrpc.php">
<!--[if IE]>
<link rel="stylesheet" type="text/css" href="https://www.r-bloggers.com/wp-content/themes/magazine-basic/iestyles.css" />
<![endif]-->

<!-- Open Graph Meta Tags generated by MashShare 3.4.4 - https://mashshare.net -->
<meta property="og:type" content="article"> 
<meta property="og:title" content="Genome-wide association studies in R">
<meta property="og:description" content="This time I elaborate on a much more specific subject that will mostly concern biologists and geneticists. I will try my best to outline the approach as to ensure non-experts will still have a basic understanding. This tutorial illustrates the power of genome-wide association (GWA) studies by mapping the genetic determinants of cholesterol levels using … Continue reading Genome-wide association studies in R →">
<meta property="og:url" content="https://www.r-bloggers.com/genome-wide-association-studies-in-r/">
<meta property="og:site_name" content="R-bloggers">
<meta property="article:publisher" content="https://www.facebook.com/rbloggers/">
<meta property="article:published_time" content="2017-10-09T11:19:28+00:00">
<meta property="article:modified_time" content="2017-10-09T11:19:28+00:00">
<meta property="og:updated_time" content="2017-10-09T11:19:28+00:00">
<!-- Open Graph Meta Tags generated by MashShare 3.4.4 - https://www.mashshare.net -->
<!-- Twitter Card generated by MashShare 3.4.4 - https://www.mashshare.net -->
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Genome-wide association studies in R">
<meta name="twitter:description" content="This time I elaborate on a much more specific subject that will mostly concern biologists and geneticists. I will try my best to outline the approach as to ensure non-experts will still have a basic understanding. This tutorial illustrates the power of genome-wide association (GWA) studies by mapping the genetic determinants of cholesterol levels using … Continue reading Genome-wide association studies in R →">
<meta name="twitter:site" content="@Rbloggers">
<!-- Twitter Card generated by MashShare 3.4.4 - https://www.mashshare.net -->


<!-- All in One SEO Pack 2.3.16 by Michael Torbert of Semper Fi Web Design[372,430] -->
<meta name="description" content="This time I elaborate on a much more specific subject that will mostly concern biologists and geneticists. I will try my best to outline the approach as to">

<link rel="canonical" href="https://www.r-bloggers.com/genome-wide-association-studies-in-r/">
<!-- /all in one seo pack -->
<link rel="dns-prefetch" href="https://s0.wp.com/">
<link rel="dns-prefetch" href="https://s.w.org/">
<!-- This site uses the Google Analytics by MonsterInsights plugin v6.2.2 - Using Analytics tracking - https://www.monsterinsights.com/ -->
<script type="text/javascript" data-cfasync="false">
		var disableStr = 'ga-disable-UA-419807-53';

	/* Function to detect opted out users */
	function __gaTrackerIsOptedOut() {
		return document.cookie.indexOf(disableStr + '=true') > -1;
	}

	/* Disable tracking if the opt-out cookie exists. */
	if ( __gaTrackerIsOptedOut() ) {
		window[disableStr] = true;
	}

	/* Opt-out function */
	function __gaTrackerOptout() {
	  document.cookie = disableStr + '=true; expires=Thu, 31 Dec 2099 23:59:59 UTC; path=/';
	  window[disableStr] = true;
	}
		(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
		(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
		m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','//www.google-analytics.com/analytics.js','__gaTracker');

	__gaTracker('create', 'UA-419807-53', 'auto');
	__gaTracker('set', 'forceSSL', true);
	__gaTracker('send','pageview');
</script>
<!-- / Google Analytics by MonsterInsights -->
		<script type="text/javascript">
			window._wpemojiSettings = {"baseUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2\/72x72\/","ext":".png","svgUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2\/svg\/","svgExt":".svg","source":{"concatemoji":"https:\/\/www.r-bloggers.com\/wp-includes\/js\/wp-emoji-release.min.js?ver=4.6.1"}};
			!function(a,b,c){function d(a){var c,d,e,f,g,h=b.createElement("canvas"),i=h.getContext&&h.getContext("2d"),j=String.fromCharCode;if(!i||!i.fillText)return!1;switch(i.textBaseline="top",i.font="600 32px Arial",a){case"flag":return i.fillText(j(55356,56806,55356,56826),0,0),!(h.toDataURL().length<3e3)&&(i.clearRect(0,0,h.width,h.height),i.fillText(j(55356,57331,65039,8205,55356,57096),0,0),c=h.toDataURL(),i.clearRect(0,0,h.width,h.height),i.fillText(j(55356,57331,55356,57096),0,0),d=h.toDataURL(),c!==d);case"diversity":return i.fillText(j(55356,57221),0,0),e=i.getImageData(16,16,1,1).data,f=e[0]+","+e[1]+","+e[2]+","+e[3],i.fillText(j(55356,57221,55356,57343),0,0),e=i.getImageData(16,16,1,1).data,g=e[0]+","+e[1]+","+e[2]+","+e[3],f!==g;case"simple":return i.fillText(j(55357,56835),0,0),0!==i.getImageData(16,16,1,1).data[0];case"unicode8":return i.fillText(j(55356,57135),0,0),0!==i.getImageData(16,16,1,1).data[0];case"unicode9":return i.fillText(j(55358,56631),0,0),0!==i.getImageData(16,16,1,1).data[0]}return!1}function e(a){var c=b.createElement("script");c.src=a,c.type="text/javascript",b.getElementsByTagName("head")[0].appendChild(c)}var f,g,h,i;for(i=Array("simple","flag","unicode8","diversity","unicode9"),c.supports={everything:!0,everythingExceptFlag:!0},h=0;h<i.length;h++)c.supports[i[h]]=d(i[h]),c.supports.everything=c.supports.everything&&c.supports[i[h]],"flag"!==i[h]&&(c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&c.supports[i[h]]);c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&!c.supports.flag,c.DOMReady=!1,c.readyCallback=function(){c.DOMReady=!0},c.supports.everything||(g=function(){c.readyCallback()},b.addEventListener?(b.addEventListener("DOMContentLoaded",g,!1),a.addEventListener("load",g,!1)):(a.attachEvent("onload",g),b.attachEvent("onreadystatechange",function(){"complete"===b.readyState&&c.readyCallback()})),f=c.source||{},f.concatemoji?e(f.concatemoji):f.wpemoji&&f.twemoji&&(e(f.twemoji),e(f.wpemoji)))}(window,document,window._wpemojiSettings);
		</script><script src="./Genome-wide association studies in R _ R-bloggers_files/wp-emoji-release.min.js.download" type="text/javascript"></script>
		<style type="text/css">
img.wp-smiley,
img.emoji {
	display: inline !important;
	border: none !important;
	box-shadow: none !important;
	height: 1em !important;
	width: 1em !important;
	margin: 0 .07em !important;
	vertical-align: -0.1em !important;
	background: none !important;
	padding: 0 !important;
}
</style>
<link rel="stylesheet" id="wgs2-css" href="./Genome-wide association studies in R _ R-bloggers_files/wgs2.css" type="text/css" media="all">
<link rel="stylesheet" id="snp_styles_reset-css" href="./Genome-wide association studies in R _ R-bloggers_files/reset.min.css" type="text/css" media="all">
<link rel="stylesheet" id="fancybox2-css" href="./Genome-wide association studies in R _ R-bloggers_files/jquery.fancybox.min.css" type="text/css" media="all">
<link rel="stylesheet" id="snp_styles_theme6-css" href="./Genome-wide association studies in R _ R-bloggers_files/theme6.css" type="text/css" media="all">
<link rel="stylesheet" id="mashsb-styles-css" href="./Genome-wide association studies in R _ R-bloggers_files/mashsb.min.css" type="text/css" media="all">
<style id="mashsb-styles-inline-css" type="text/css">
.mashsb-count {color:#cccccc;}@media only screen and (min-width:568px){.mashsb-buttons a {min-width: 177px;}}
</style>
<link rel="stylesheet" id="wpt-twitter-feed-css" href="./Genome-wide association studies in R _ R-bloggers_files/twitter-feed.css" type="text/css" media="all">
<link rel="stylesheet" id="jetpack_css-css" href="./Genome-wide association studies in R _ R-bloggers_files/jetpack.css" type="text/css" media="all">
<link rel="stylesheet" id="codebox-css" href="./Genome-wide association studies in R _ R-bloggers_files/codebox.css" type="text/css" media="screen">
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/jquery.min.js.download"></script>
<script type="text/javascript">try{jQuery.noConflict();}catch(e){};</script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/jquery-migrate.min.js.download"></script>
<script type="text/javascript">
/* <![CDATA[ */
var related_posts_js_options = {"post_heading":"h4"};
/* ]]> */
</script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/related-posts.js.download"></script>
<script type="text/javascript">
/* <![CDATA[ */
var monsterinsights_frontend = {"js_events_tracking":"true","is_debug_mode":"false","download_extensions":"doc,exe,js,pdf,ppt,tgz,zip,xls","inbound_paths":"","home_url":"https:\/\/www.r-bloggers.com","track_download_as":"event","internal_label":"int","hash_tracking":"false"};
/* ]]> */
</script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/frontend.min.js.download"></script>
<script type="text/javascript">
/* <![CDATA[ */
var mashsb = {"shares":"20","round_shares":"1","animate_shares":"0","dynamic_buttons":"0","share_url":"https:\/\/www.r-bloggers.com\/genome-wide-association-studies-in-r\/","title":"Genome-wide+association+studies+in+R","image":null,"desc":"This time I elaborate on a much more specific subject that will mostly concern biologists and geneticists. I will try my best to outline the approach as to ensure non-experts will still have a basic understanding. This tutorial illustrates the power of genome-wide association (GWA) studies by mapping the genetic determinants of cholesterol levels using \u2026 Continue reading Genome-wide association studies in\u00a0R \u2192\n","hashtag":"Rbloggers","subscribe":"link","subscribe_url":"https:\/\/feedburner.google.com\/fb\/a\/mailverify?uri=RBloggers","activestatus":"1","singular":"1","twitter_popup":"1","refresh":"0"};
/* ]]> */
</script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/mashsb.min.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/codebox.js.download"></script>
<link rel="https://api.w.org/" href="https://www.r-bloggers.com/wp-json/">
<link rel="EditURI" type="application/rsd+xml" title="RSD" href="https://www.r-bloggers.com/xmlrpc.php?rsd">
<link rel="wlwmanifest" type="application/wlwmanifest+xml" href="https://www.r-bloggers.com/wp-includes/wlwmanifest.xml"> 
<link rel="prev" title="RStudio v1.1 Released" href="https://www.r-bloggers.com/rstudio-v1-1-released/">
<meta name="generator" content="WordPress 4.6.1">
<link rel="shortlink" href="https://wp.me/pMm6L-Ey0">
<link rel="alternate" type="application/json+oembed" href="https://www.r-bloggers.com/wp-json/oembed/1.0/embed?url=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F">
<link rel="alternate" type="text/xml+oembed" href="https://www.r-bloggers.com/wp-json/oembed/1.0/embed?url=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F&amp;format=xml">
<!-- This site is powered by Tweet, Like, Plusone and Share Plugin - http://techxt.com/tweet-like-google-1-and-share-plugin-wordpress/ -->
<style type="text/css">div.socialicons{float:left;display:block;margin-right: 10px;line-height: 1;}div.socialiconsv{line-height: 1;}div.socialicons p{margin-bottom: 0px !important;margin-top: 0px !important;padding-bottom: 0px !important;padding-top: 0px !important;}div.social4iv{background: none repeat scroll 0 0 #FFFFFF;border: 1px solid #aaa;border-radius: 3px 3px 3px 3px;box-shadow: 3px 3px 3px #DDDDDD;padding: 3px;position: fixed;text-align: center;top: 55px;width: 68px;display:none;}div.socialiconsv{padding-bottom: 5px;}</style>

<link rel="dns-prefetch" href="https://i0.wp.com/">
<link rel="dns-prefetch" href="https://i1.wp.com/">
<link rel="dns-prefetch" href="https://i2.wp.com/">
<style type="text/css">img#wpstats{display:none}</style><link rel="amphtml" href="https://www.r-bloggers.com/genome-wide-association-studies-in-r/amp/"><link rel="stylesheet" type="text/css" href="./Genome-wide association studies in R _ R-bloggers_files/shCore.css"><link rel="stylesheet" type="text/css" href="./Genome-wide association studies in R _ R-bloggers_files/shThemeDefault.css"><style type="text/css" id="syntaxhighlighteranchor"></style>

<!-- WordPress Facebook Open Graph protocol plugin (WPFBOGP v2.0.13) http://rynoweb.com/wordpress-plugins/ -->
<meta property="fb:app_id" content="143207055763488">
<meta property="og:image" content="https://www.r-bloggers.com/wp-content/uploads/2016/08/R_single_01-200-1.png">
<meta property="og:image" content="https://pixel.wp.com/b.gif?host=poissonisfish.wordpress.com&amp;blog=120365233&amp;%23038;post=3115&amp;%23038;subd=poissonisfish&amp;%23038;ref=&amp;%23038;feed=1">
<meta property="og:image" content="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=1108">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=1108">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=1108">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=-%5Clog_%7B10%7D+%28P%29+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=%5Cbeta+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=1108">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=2pq+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=%7CF%7C+%3D+1+-+%5Cfrac%7BH%7D%7B2pq%7D+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=2">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=2pq+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=p%5E2+%2B+2pq+%2B+q%5E2+%3D+1+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=2">
<meta property="og:image" content="https://s0.wp.com/latex.php?latex=P%28%5Cbeta%7CH_0%3A+%5Cbeta%3D0%29+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1">
<meta property="og:image" content="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=1108">
<!-- // end wpfbogp -->
<link rel="icon" href="https://i2.wp.com/www.r-bloggers.com/wp-content/uploads/2016/08/cropped-R_single_01-200.png?fit=32%2C32&amp;ssl=1" sizes="32x32">
<link rel="icon" href="https://i2.wp.com/www.r-bloggers.com/wp-content/uploads/2016/08/cropped-R_single_01-200.png?fit=192%2C192&amp;ssl=1" sizes="192x192">
<link rel="apple-touch-icon-precomposed" href="https://i2.wp.com/www.r-bloggers.com/wp-content/uploads/2016/08/cropped-R_single_01-200.png?fit=180%2C180&amp;ssl=1">
<meta name="msapplication-TileImage" content="https://i2.wp.com/www.r-bloggers.com/wp-content/uploads/2016/08/cropped-R_single_01-200.png?fit=270%2C270&amp;ssl=1">
<script src="./Genome-wide association studies in R _ R-bloggers_files/secureAnonymousFramework"></script><style type="text/css">.fb_hidden{position:absolute;top:-10000px;z-index:10001}.fb_reposition{overflow:hidden;position:relative}.fb_invisible{display:none}.fb_reset{background:none;border:0;border-spacing:0;color:#000;cursor:auto;direction:ltr;font-family:"lucida grande", tahoma, verdana, arial, sans-serif;font-size:11px;font-style:normal;font-variant:normal;font-weight:normal;letter-spacing:normal;line-height:1;margin:0;overflow:visible;padding:0;text-align:left;text-decoration:none;text-indent:0;text-shadow:none;text-transform:none;visibility:visible;white-space:normal;word-spacing:normal}.fb_reset>div{overflow:hidden}.fb_link img{border:none}@keyframes fb_transform{from{opacity:0;transform:scale(.95)}to{opacity:1;transform:scale(1)}}.fb_animate{animation:fb_transform .3s forwards}.fb_shrink_active{opacity:1;transform:scale(1, 1);transition-duration:200ms;transition-timing-function:ease-out}.fb_shrink_active:active{opacity:.5;transform:scale(.75, .75)}
.fb_dialog{background:rgba(82, 82, 82, .7);position:absolute;top:-10000px;z-index:10001}.fb_reset .fb_dialog_legacy{overflow:visible}.fb_dialog_advanced{padding:10px;-moz-border-radius:8px;-webkit-border-radius:8px;border-radius:8px}.fb_dialog_content{background:#fff;color:#333}.fb_dialog_close_icon{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/yq/r/IE9JII6Z1Ys.png) no-repeat scroll 0 0 transparent;cursor:pointer;display:block;height:15px;position:absolute;right:18px;top:17px;width:15px}.fb_dialog_mobile .fb_dialog_close_icon{top:5px;left:5px;right:auto}.fb_dialog_padding{background-color:transparent;position:absolute;width:1px;z-index:-1}.fb_dialog_close_icon:hover{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/yq/r/IE9JII6Z1Ys.png) no-repeat scroll 0 -15px transparent}.fb_dialog_close_icon:active{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/yq/r/IE9JII6Z1Ys.png) no-repeat scroll 0 -30px transparent}.fb_dialog_loader{background-color:#f6f7f9;border:1px solid #606060;font-size:24px;padding:20px}.fb_dialog_top_left,.fb_dialog_top_right,.fb_dialog_bottom_left,.fb_dialog_bottom_right{height:10px;width:10px;overflow:hidden;position:absolute}.fb_dialog_top_left{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/ye/r/8YeTNIlTZjm.png) no-repeat 0 0;left:-10px;top:-10px}.fb_dialog_top_right{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/ye/r/8YeTNIlTZjm.png) no-repeat 0 -10px;right:-10px;top:-10px}.fb_dialog_bottom_left{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/ye/r/8YeTNIlTZjm.png) no-repeat 0 -20px;bottom:-10px;left:-10px}.fb_dialog_bottom_right{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/ye/r/8YeTNIlTZjm.png) no-repeat 0 -30px;right:-10px;bottom:-10px}.fb_dialog_vert_left,.fb_dialog_vert_right,.fb_dialog_horiz_top,.fb_dialog_horiz_bottom{position:absolute;background:#525252;filter:alpha(opacity=70);opacity:.7}.fb_dialog_vert_left,.fb_dialog_vert_right{width:10px;height:100%}.fb_dialog_vert_left{margin-left:-10px}.fb_dialog_vert_right{right:0;margin-right:-10px}.fb_dialog_horiz_top,.fb_dialog_horiz_bottom{width:100%;height:10px}.fb_dialog_horiz_top{margin-top:-10px}.fb_dialog_horiz_bottom{bottom:0;margin-bottom:-10px}.fb_dialog_iframe{line-height:0}.fb_dialog_content .dialog_title{background:#6d84b4;border:1px solid #365899;color:#fff;font-size:14px;font-weight:bold;margin:0}.fb_dialog_content .dialog_title>span{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/yd/r/Cou7n-nqK52.gif) no-repeat 5px 50%;float:left;padding:5px 0 7px 26px}body.fb_hidden{-webkit-transform:none;height:100%;margin:0;overflow:visible;position:absolute;top:-10000px;left:0;width:100%}.fb_dialog.fb_dialog_mobile.loading{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/ya/r/3rhSv5V8j3o.gif) white no-repeat 50% 50%;min-height:100%;min-width:100%;overflow:hidden;position:absolute;top:0;z-index:10001}.fb_dialog.fb_dialog_mobile.loading.centered{width:auto;height:auto;min-height:initial;min-width:initial;background:none}.fb_dialog.fb_dialog_mobile.loading.centered #fb_dialog_loader_spinner{width:100%}.fb_dialog.fb_dialog_mobile.loading.centered .fb_dialog_content{background:none}.loading.centered #fb_dialog_loader_close{color:#fff;display:block;padding-top:20px;clear:both;font-size:18px}#fb-root #fb_dialog_ipad_overlay{background:rgba(0, 0, 0, .45);position:absolute;bottom:0;left:0;right:0;top:0;width:100%;min-height:100%;z-index:10000}#fb-root #fb_dialog_ipad_overlay.hidden{display:none}.fb_dialog.fb_dialog_mobile.loading iframe{visibility:hidden}.fb_dialog_content .dialog_header{-webkit-box-shadow:white 0 1px 1px -1px inset;background:-webkit-gradient(linear, 0% 0%, 0% 100%, from(#738ABA), to(#2C4987));border-bottom:1px solid;border-color:#1d4088;color:#fff;font:14px Helvetica, sans-serif;font-weight:bold;text-overflow:ellipsis;text-shadow:rgba(0, 30, 84, .296875) 0 -1px 0;vertical-align:middle;white-space:nowrap}.fb_dialog_content .dialog_header table{-webkit-font-smoothing:subpixel-antialiased;height:43px;width:100%}.fb_dialog_content .dialog_header td.header_left{font-size:12px;padding-left:5px;vertical-align:middle;width:60px}.fb_dialog_content .dialog_header td.header_right{font-size:12px;padding-right:5px;vertical-align:middle;width:60px}.fb_dialog_content .touchable_button{background:-webkit-gradient(linear, 0% 0%, 0% 100%, from(#4966A6), color-stop(.5, #355492), to(#2A4887));border:1px solid #29487d;-webkit-background-clip:padding-box;-webkit-border-radius:3px;-webkit-box-shadow:rgba(0, 0, 0, .117188) 0 1px 1px inset, rgba(255, 255, 255, .167969) 0 1px 0;display:inline-block;margin-top:3px;max-width:85px;line-height:18px;padding:4px 12px;position:relative}.fb_dialog_content .dialog_header .touchable_button input{border:none;background:none;color:#fff;font:12px Helvetica, sans-serif;font-weight:bold;margin:2px -12px;padding:2px 6px 3px 6px;text-shadow:rgba(0, 30, 84, .296875) 0 -1px 0}.fb_dialog_content .dialog_header .header_center{color:#fff;font-size:16px;font-weight:bold;line-height:18px;text-align:center;vertical-align:middle}.fb_dialog_content .dialog_content{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/y9/r/jKEcVPZFk-2.gif) no-repeat 50% 50%;border:1px solid #555;border-bottom:0;border-top:0;height:150px}.fb_dialog_content .dialog_footer{background:#f6f7f9;border:1px solid #555;border-top-color:#ccc;height:40px}#fb_dialog_loader_close{float:left}.fb_dialog.fb_dialog_mobile .fb_dialog_close_button{text-shadow:rgba(0, 30, 84, .296875) 0 -1px 0}.fb_dialog.fb_dialog_mobile .fb_dialog_close_icon{visibility:hidden}#fb_dialog_loader_spinner{animation:rotateSpinner 1.2s linear infinite;background-color:transparent;background-image:url(http://static.xx.fbcdn.net/rsrc.php/v3/yD/r/t-wz8gw1xG1.png);background-repeat:no-repeat;background-position:50% 50%;height:24px;width:24px}@keyframes rotateSpinner{0%{transform:rotate(0deg)}100%{transform:rotate(360deg)}}
.fb_iframe_widget{display:inline-block;position:relative}.fb_iframe_widget span{display:inline-block;position:relative;text-align:justify}.fb_iframe_widget iframe{position:absolute}.fb_iframe_widget_fluid_desktop,.fb_iframe_widget_fluid_desktop span,.fb_iframe_widget_fluid_desktop iframe{max-width:100%}.fb_iframe_widget_fluid_desktop iframe{min-width:220px;position:relative}.fb_iframe_widget_lift{z-index:1}.fb_hide_iframes iframe{position:relative;left:-10000px}.fb_iframe_widget_loader{position:relative;display:inline-block}.fb_iframe_widget_fluid{display:inline}.fb_iframe_widget_fluid span{width:100%}.fb_iframe_widget_loader iframe{min-height:32px;z-index:2;zoom:1}.fb_iframe_widget_loader .FB_Loader{background:url(http://static.xx.fbcdn.net/rsrc.php/v3/y9/r/jKEcVPZFk-2.gif) no-repeat;height:32px;width:32px;margin-left:-16px;position:absolute;left:50%;z-index:4}</style><script type="text/x-mathjax-config;executed=true">MathJax.Hub.Config({
                    jax: ["input/TeX","input/MathML","input/AsciiMath","output/CommonHTML"],
  tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}
});

// Renders math images using MathJax instead
// which makes text selectable
MathJax.Extension.myImg2jax = {
  version: "1.0",

  PreProcess: function (element) {
    var images = element.getElementsByTagName("img");
    for (var i = images.length - 1; i >= 0; i--) {
    var img = images[i];
    if (img.className === "mwe-math-fallback-image-inline") {
        var script = document.createElement("script");
        script.type = "math/tex";
        MathJax.HTML.setScript(script, img.alt);
        img.parentNode.replaceChild(script, img);
      }
    }
  }
};

MathJax.Hub.Register.PreProcessor(["PreProcess", MathJax.Extension.myImg2jax]);</script><script src="./Genome-wide association studies in R _ R-bloggers_files/MathJax.js.download" id=""></script><script src="./Genome-wide association studies in R _ R-bloggers_files/jsapi" type="text/javascript"></script><style type="text/css">* html #li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link{height:1% !important;}#li_ui_li_gen_1507763900872_0{position:relative !important;overflow:visible !important;display:block !important;}#li_ui_li_gen_1507763900872_0 span{-webkit-box-sizing:content-box !important;-moz-box-sizing:content-box !important;box-sizing:content-box !important;}#li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link{border:0 !important;height:20px !important;text-decoration:none !important;padding:0 !important;margin:0 !important;display:inline-block !important;}#li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link:link, #li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link:visited, #li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link:hover, #li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link:active{border:0 !important;text-decoration:none !important;}#li_ui_li_gen_1507763900872_0 a#li_ui_li_gen_1507763900872_0-link:after{content:"." !important;display:block !important;clear:both !important;visibility:hidden !important;line-height:0 !important;height:0 !important;}#li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-logo{background-image:url(https://static.licdn.com/scds/common/u/images/apps/connect/sprites/sprite_connect_v14.png) !important;background-position:-56px -423px !important;background-repeat:no-repeat !important;background-color:#0077b5 !important;background-size:135px auto !important;cursor:pointer !important;border:0 !important;border-right:1px solid #066094 !important;text-indent:-9999em !important;overflow:hidden !important;padding:0 !important;margin:0 !important;position:absolute !important;left:0px !important;top:0px !important;display:block !important;width:20px !important;height:20px !important;float:right !important;border-radius:2px !important;-webkit-border-radius:2px !important;border-top-right-radius:2px !important;border-bottom-right-radius:2px !important;-webkit-border-top-right-radius:2px !important;-webkit-border-bottom-right-radius:2px !important;}#li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title{color:#fff !important;cursor:pointer !important;display:block !important;white-space:nowrap !important;float:left !important;margin-left:1px !important;vertical-align:top !important;overflow:hidden !important;text-align:center !important;height:18px !important;padding:0 4px 0 23px !important;border:1px solid #000 !important;border-top-color:#0077b5 !important;border-right-color:#0077b5 !important;border-bottom-color:#0077b5 !important;border-left-color:#0077b5 !important;text-shadow:0 -1px #005887 !important;line-height:20px !important;border-radius:2px !important;-webkit-border-radius:2px !important;border-top-right-radius:2px !important;border-bottom-right-radius:2px !important;-webkit-border-top-right-radius:2px !important;-webkit-border-bottom-right-radius:2px !important;background-color:#0077b5 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0077b5), color-stop(100%,#0077b5)) !important; background-image: -webkit-linear-gradient(top, #0077b5 0%, #0077b5 100%) !important;}#li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title{border:1px solid #000 !important;border-top-color:#0369a0 !important;border-right-color:#0369a0 !important;border-bottom-color:#0369a0 !important;border-left-color:#0369a0 !important;background-color:#0369a0 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0369a0), color-stop(100%,#0369a0)) !important; background-image: -webkit-linear-gradient(top, #0369a0 0%, #0369a0 100%) !important;}#li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title, #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title{color:#fff !important;border:1px solid #000 !important;border-top-color:#066094 !important;border-right-color:#066094 !important;border-bottom-color:#066094 !important;border-left-color:#066094 !important;background-color:#066094 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#066094), color-stop(100%,#066094)) !important; background-image: -webkit-linear-gradient(top, #066094 0%, #066094 100%) !important;}.IN-shadowed #li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title{}.IN-shadowed #li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title{}.IN-shadowed #li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title, .IN-shadowed #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title{}#li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title-text, #li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title-text *{color:#fff !important;font-size:11px !important;font-family:Arial, sans-serif !important;font-weight:bold !important;font-style:normal !important;-webkit-font-smoothing:antialiased !important;display:inline-block !important;background:transparent none !important;vertical-align:top !important;height:18px !important;line-height:20px !important;float:none !important;}#li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title-text strong{font-weight:bold !important;}#li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title-text em{font-style:italic !important;}#li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title-text, #li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title-text *{color:#fff !important;}#li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title-text, #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title-text, #li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title-text *, #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title-text *{color:#fff !important;}#li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title #li_ui_li_gen_1507763900872_0-mark{display:inline-block !important;width:0px !important;overflow:hidden !important;}.success #li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title{color:#fff !important;border-top-color:#0077b5 !important;border-right-color:#0077b5 !important;border-bottom-color:#0077b5 !important;border-left-color:#0077b5 !important;background-color:#0077b5 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0077b5), color-stop(100%,#0077b5)) !important; background-image: -webkit-linear-gradient(top, #0077b5 0%, #0077b5 100%) !important;}.success #li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title-text, .success #li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title-text *{color:#fff !important;}.IN-shadowed .success #li_ui_li_gen_1507763900872_0 #li_ui_li_gen_1507763900872_0-title{}.success #li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title{color:#fff !important;border-top-color:#0369a0 !important;border-right-color:#0369a0 !important;border-bottom-color:#0369a0 !important;border-left-color:#0369a0 !important;background-color:#0369a0 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0369a0), color-stop(100%,#0369a0)) !important; background-image: -webkit-linear-gradient(top, #0369a0 0%, #0369a0 100%) !important;}.success #li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title-text, .success #li_ui_li_gen_1507763900872_0.hovered #li_ui_li_gen_1507763900872_0-title-text *{color:#fff !important;}.success #li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title, .success #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title{color:#fff !important;border-top-color:#066094 !important;border-right-color:#066094 !important;border-bottom-color:#066094 !important;border-left-color:#066094 !important;background-color:#066094 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#066094), color-stop(100%,#066094)) !important; background-image: -webkit-linear-gradient(top, #066094 0%, #066094 100%) !important;}.success #li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title-text, .success #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title-text, .success #li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title-text *, .success #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title-text *{color:#fff !important;}.IN-shadowed .success #li_ui_li_gen_1507763900872_0.clicked #li_ui_li_gen_1507763900872_0-title, .IN-shadowed .success #li_ui_li_gen_1507763900872_0.down #li_ui_li_gen_1507763900872_0-title{}#li_ui_li_gen_1507763900889_1-container.IN-right {display:inline-block !important;float:left !important;overflow:visible !important;position:relative !important;height:20px !important;padding-left:2px !important;line-height:1px !important;cursor:pointer !important;}#li_ui_li_gen_1507763900889_1.IN-right {display:block !important;float:left !important;height:20px !important;margin-right:4px !important;padding-right:4px !important;background-image:url(https://static.licdn.com/scds/common/u/images/apps/connect/sprites/sprite_connect_v14.png) !important;background-color:transparent !important;background-repeat:no-repeat !important;background-position:right -447px !important;}#li_ui_li_gen_1507763900889_1-inner.IN-right {display:block !important;float:left !important;padding-left:8px !important;text-align:center !important;background-image:url(https://static.licdn.com/scds/common/u/images/apps/connect/sprites/sprite_connect_v14.png) !important;background-color:transparent !important;background-repeat:no-repeat !important;background-position:0px -426px !important;}#li_ui_li_gen_1507763900889_1-content.IN-right {display:inline !important;font-size:11px !important;color:#4e4e4e !important;font-weight:bold !important;font-family:Arial, sans-serif !important;line-height:20px !important;padding:0 5px 0 5px !important;}#li_ui_li_gen_1507763900889_1-container.IN-empty {display:none !important;}#li_ui_li_gen_1507763900889_1-container.IN-hidden #li_ui_li_gen_1507763900889_1 {display:none !important;}* html #li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link{height:1% !important;}#li_ui_li_gen_1507763900895_2{position:relative !important;overflow:visible !important;display:block !important;}#li_ui_li_gen_1507763900895_2 span{-webkit-box-sizing:content-box !important;-moz-box-sizing:content-box !important;box-sizing:content-box !important;}#li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link{border:0 !important;height:20px !important;text-decoration:none !important;padding:0 !important;margin:0 !important;display:inline-block !important;}#li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link:link, #li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link:visited, #li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link:hover, #li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link:active{border:0 !important;text-decoration:none !important;}#li_ui_li_gen_1507763900895_2 a#li_ui_li_gen_1507763900895_2-link:after{content:"." !important;display:block !important;clear:both !important;visibility:hidden !important;line-height:0 !important;height:0 !important;}#li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-logo{background-image:url(https://static.licdn.com/scds/common/u/images/apps/connect/sprites/sprite_connect_v14.png) !important;background-position:-56px -423px !important;background-repeat:no-repeat !important;background-color:#0077b5 !important;background-size:135px auto !important;cursor:pointer !important;border:0 !important;border-right:1px solid #066094 !important;text-indent:-9999em !important;overflow:hidden !important;padding:0 !important;margin:0 !important;position:absolute !important;left:0px !important;top:0px !important;display:block !important;width:20px !important;height:20px !important;float:right !important;border-radius:2px !important;-webkit-border-radius:2px !important;border-top-right-radius:2px !important;border-bottom-right-radius:2px !important;-webkit-border-top-right-radius:2px !important;-webkit-border-bottom-right-radius:2px !important;}#li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title{color:#fff !important;cursor:pointer !important;display:block !important;white-space:nowrap !important;float:left !important;margin-left:1px !important;vertical-align:top !important;overflow:hidden !important;text-align:center !important;height:18px !important;padding:0 4px 0 23px !important;border:1px solid #000 !important;border-top-color:#0077b5 !important;border-right-color:#0077b5 !important;border-bottom-color:#0077b5 !important;border-left-color:#0077b5 !important;text-shadow:0 -1px #005887 !important;line-height:20px !important;border-radius:2px !important;-webkit-border-radius:2px !important;border-top-right-radius:2px !important;border-bottom-right-radius:2px !important;-webkit-border-top-right-radius:2px !important;-webkit-border-bottom-right-radius:2px !important;background-color:#0077b5 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0077b5), color-stop(100%,#0077b5)) !important; background-image: -webkit-linear-gradient(top, #0077b5 0%, #0077b5 100%) !important;}#li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title{border:1px solid #000 !important;border-top-color:#0369a0 !important;border-right-color:#0369a0 !important;border-bottom-color:#0369a0 !important;border-left-color:#0369a0 !important;background-color:#0369a0 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0369a0), color-stop(100%,#0369a0)) !important; background-image: -webkit-linear-gradient(top, #0369a0 0%, #0369a0 100%) !important;}#li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title, #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title{color:#fff !important;border:1px solid #000 !important;border-top-color:#066094 !important;border-right-color:#066094 !important;border-bottom-color:#066094 !important;border-left-color:#066094 !important;background-color:#066094 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#066094), color-stop(100%,#066094)) !important; background-image: -webkit-linear-gradient(top, #066094 0%, #066094 100%) !important;}.IN-shadowed #li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title{}.IN-shadowed #li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title{}.IN-shadowed #li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title, .IN-shadowed #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title{}#li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title-text, #li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title-text *{color:#fff !important;font-size:11px !important;font-family:Arial, sans-serif !important;font-weight:bold !important;font-style:normal !important;-webkit-font-smoothing:antialiased !important;display:inline-block !important;background:transparent none !important;vertical-align:top !important;height:18px !important;line-height:20px !important;float:none !important;}#li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title-text strong{font-weight:bold !important;}#li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title-text em{font-style:italic !important;}#li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title-text, #li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title-text *{color:#fff !important;}#li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title-text, #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title-text, #li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title-text *, #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title-text *{color:#fff !important;}#li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title #li_ui_li_gen_1507763900895_2-mark{display:inline-block !important;width:0px !important;overflow:hidden !important;}.success #li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title{color:#fff !important;border-top-color:#0077b5 !important;border-right-color:#0077b5 !important;border-bottom-color:#0077b5 !important;border-left-color:#0077b5 !important;background-color:#0077b5 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0077b5), color-stop(100%,#0077b5)) !important; background-image: -webkit-linear-gradient(top, #0077b5 0%, #0077b5 100%) !important;}.success #li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title-text, .success #li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title-text *{color:#fff !important;}.IN-shadowed .success #li_ui_li_gen_1507763900895_2 #li_ui_li_gen_1507763900895_2-title{}.success #li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title{color:#fff !important;border-top-color:#0369a0 !important;border-right-color:#0369a0 !important;border-bottom-color:#0369a0 !important;border-left-color:#0369a0 !important;background-color:#0369a0 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#0369a0), color-stop(100%,#0369a0)) !important; background-image: -webkit-linear-gradient(top, #0369a0 0%, #0369a0 100%) !important;}.success #li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title-text, .success #li_ui_li_gen_1507763900895_2.hovered #li_ui_li_gen_1507763900895_2-title-text *{color:#fff !important;}.success #li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title, .success #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title{color:#fff !important;border-top-color:#066094 !important;border-right-color:#066094 !important;border-bottom-color:#066094 !important;border-left-color:#066094 !important;background-color:#066094 !important;background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0%,#066094), color-stop(100%,#066094)) !important; background-image: -webkit-linear-gradient(top, #066094 0%, #066094 100%) !important;}.success #li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title-text, .success #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title-text, .success #li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title-text *, .success #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title-text *{color:#fff !important;}.IN-shadowed .success #li_ui_li_gen_1507763900895_2.clicked #li_ui_li_gen_1507763900895_2-title, .IN-shadowed .success #li_ui_li_gen_1507763900895_2.down #li_ui_li_gen_1507763900895_2-title{}#li_ui_li_gen_1507763900898_3-container.IN-right {display:inline-block !important;float:left !important;overflow:visible !important;position:relative !important;height:20px !important;padding-left:2px !important;line-height:1px !important;cursor:pointer !important;}#li_ui_li_gen_1507763900898_3.IN-right {display:block !important;float:left !important;height:20px !important;margin-right:4px !important;padding-right:4px !important;background-image:url(https://static.licdn.com/scds/common/u/images/apps/connect/sprites/sprite_connect_v14.png) !important;background-color:transparent !important;background-repeat:no-repeat !important;background-position:right -447px !important;}#li_ui_li_gen_1507763900898_3-inner.IN-right {display:block !important;float:left !important;padding-left:8px !important;text-align:center !important;background-image:url(https://static.licdn.com/scds/common/u/images/apps/connect/sprites/sprite_connect_v14.png) !important;background-color:transparent !important;background-repeat:no-repeat !important;background-position:0px -426px !important;}#li_ui_li_gen_1507763900898_3-content.IN-right {display:inline !important;font-size:11px !important;color:#4e4e4e !important;font-weight:bold !important;font-family:Arial, sans-serif !important;line-height:20px !important;padding:0 5px 0 5px !important;}#li_ui_li_gen_1507763900898_3-container.IN-empty {display:none !important;}#li_ui_li_gen_1507763900898_3-container.IN-hidden #li_ui_li_gen_1507763900898_3 {display:none !important;}</style><link type="text/css" href="./Genome-wide association studies in R _ R-bloggers_files/default+en.css" rel="stylesheet"><link type="text/css" href="./Genome-wide association studies in R _ R-bloggers_files/default.css" rel="stylesheet"><script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/default+en.I.js.download"></script><style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 2px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 2px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: 1em}
.MathJax_MenuRadioCheck.RTL {right: 1em; left: auto}
.MathJax_MenuLabel {padding: 2px 2em 4px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #CCCCCC; margin: 4px 1px 0px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: Highlight; color: HighlightText}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css">.MathJax_Preview .MJXf-math {color: inherit!important}
</style><style type="text/css">.MJX_Assistive_MathML {position: absolute!important; top: 0; left: 0; clip: rect(1px, 1px, 1px, 1px); padding: 1px 0 0 0!important; border: 0!important; height: 1px!important; width: 1px!important; overflow: hidden!important; display: block!important; -webkit-touch-callout: none; -webkit-user-select: none; -khtml-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none}
.MJX_Assistive_MathML.MJX_Assistive_MathML_Block {width: 100%!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css">.gsc-control-cse{font-family:Arial, sans-serif;border-color:#FFFFFF;background-color:#FFFFFF}.gsc-control-cse .gsc-table-result{font-family:Arial, sans-serif}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#D9D9D9}input.gsc-search-button,input.gsc-search-button:hover,input.gsc-search-button:focus{border-color:#666666;background-color:#CECECE;background-image:none;filter:none}.gsc-tabHeader.gsc-tabhInactive{border-color:#E9E9E9;background-color:#E9E9E9}.gsc-tabHeader.gsc-tabhActive{border-color:#FF9900;border-bottom-color:#FFFFFF;background-color:#FFFFFF}.gsc-tabsArea{border-color:#FF9900}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#FFFFFF;background-color:#FFFFFF}.gsc-webResult.gsc-result:hover,.gsc-imageResult:hover{border-color:#FFFFFF;background-color:#FFFFFF}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult .gs-snippet,.gs-imageResult .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl,.gs-imageResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gs-webResult div.gs-visibleUrl-short{display:none}.gs-webResult div.gs-visibleUrl-long{display:block}.gs-promotion div.gs-visibleUrl-short{display:none}.gs-promotion div.gs-visibleUrl-long{display:block}.gsc-cursor-box{border-color:#FFFFFF}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#E9E9E9;background-color:#FFFFFF;color:#0000CC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#FF9900;background-color:#FFFFFF;color:#0000CC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gsq_a{padding:0}.gscsep_a{display:none}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gsfe_a{border:1px solid #b9b9b9;border-top-color:#a0a0a0;box-shadow:inset 0px 1px 2px rgba(0,0,0,0.1);-moz-box-shadow:inset 0px 1px 2px rgba(0,0,0,0.1);-webkit-box-shadow:inset 0px 1px 2px rgba(0,0,0,0.1);}.gsfe_b{border:1px solid #4d90fe;outline:none;box-shadow:inset 0px 1px 2px rgba(0,0,0,0.3);-moz-box-shadow:inset 0px 1px 2px rgba(0,0,0,0.3);-webkit-box-shadow:inset 0px 1px 2px rgba(0,0,0,0.3);}.gssb_a{padding:0 9px}.gsib_a{padding-right:8px;padding-left:8px}.gsst_a{padding-top:5.5px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style><style type="text/css">.MJXp-script {font-size: .8em}
.MJXp-right {-webkit-transform-origin: right; -moz-transform-origin: right; -ms-transform-origin: right; -o-transform-origin: right; transform-origin: right}
.MJXp-bold {font-weight: bold}
.MJXp-italic {font-style: italic}
.MJXp-scr {font-family: MathJax_Script,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-frak {font-family: MathJax_Fraktur,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-sf {font-family: MathJax_SansSerif,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-cal {font-family: MathJax_Caligraphic,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-mono {font-family: MathJax_Typewriter,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-largeop {font-size: 150%}
.MJXp-largeop.MJXp-int {vertical-align: -.2em}
.MJXp-math {display: inline-block; line-height: 1.2; text-indent: 0; font-family: 'Times New Roman',Times,STIXGeneral,serif; white-space: nowrap; border-collapse: collapse}
.MJXp-display {display: block; text-align: center; margin: 1em 0}
.MJXp-math span {display: inline-block}
.MJXp-box {display: block!important; text-align: center}
.MJXp-box:after {content: " "}
.MJXp-rule {display: block!important; margin-top: .1em}
.MJXp-char {display: block!important}
.MJXp-mo {margin: 0 .15em}
.MJXp-mfrac {margin: 0 .125em; vertical-align: .25em}
.MJXp-denom {display: inline-table!important; width: 100%}
.MJXp-denom > * {display: table-row!important}
.MJXp-surd {vertical-align: top}
.MJXp-surd > * {display: block!important}
.MJXp-script-box > *  {display: table!important; height: 50%}
.MJXp-script-box > * > * {display: table-cell!important; vertical-align: top}
.MJXp-script-box > *:last-child > * {vertical-align: bottom}
.MJXp-script-box > * > * > * {display: block!important}
.MJXp-mphantom {visibility: hidden}
.MJXp-munderover {display: inline-table!important}
.MJXp-over {display: inline-block!important; text-align: center}
.MJXp-over > * {display: block!important}
.MJXp-munderover > * {display: table-row!important}
.MJXp-mtable {vertical-align: .25em; margin: 0 .125em}
.MJXp-mtable > * {display: inline-table!important; vertical-align: middle}
.MJXp-mtr {display: table-row!important}
.MJXp-mtd {display: table-cell!important; text-align: center; padding: .5em 0 0 .5em}
.MJXp-mtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-mlabeledtr {display: table-row!important}
.MJXp-mlabeledtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mlabeledtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MJXp-scale0 {-webkit-transform: scaleX(.0); -moz-transform: scaleX(.0); -ms-transform: scaleX(.0); -o-transform: scaleX(.0); transform: scaleX(.0)}
.MJXp-scale1 {-webkit-transform: scaleX(.1); -moz-transform: scaleX(.1); -ms-transform: scaleX(.1); -o-transform: scaleX(.1); transform: scaleX(.1)}
.MJXp-scale2 {-webkit-transform: scaleX(.2); -moz-transform: scaleX(.2); -ms-transform: scaleX(.2); -o-transform: scaleX(.2); transform: scaleX(.2)}
.MJXp-scale3 {-webkit-transform: scaleX(.3); -moz-transform: scaleX(.3); -ms-transform: scaleX(.3); -o-transform: scaleX(.3); transform: scaleX(.3)}
.MJXp-scale4 {-webkit-transform: scaleX(.4); -moz-transform: scaleX(.4); -ms-transform: scaleX(.4); -o-transform: scaleX(.4); transform: scaleX(.4)}
.MJXp-scale5 {-webkit-transform: scaleX(.5); -moz-transform: scaleX(.5); -ms-transform: scaleX(.5); -o-transform: scaleX(.5); transform: scaleX(.5)}
.MJXp-scale6 {-webkit-transform: scaleX(.6); -moz-transform: scaleX(.6); -ms-transform: scaleX(.6); -o-transform: scaleX(.6); transform: scaleX(.6)}
.MJXp-scale7 {-webkit-transform: scaleX(.7); -moz-transform: scaleX(.7); -ms-transform: scaleX(.7); -o-transform: scaleX(.7); transform: scaleX(.7)}
.MJXp-scale8 {-webkit-transform: scaleX(.8); -moz-transform: scaleX(.8); -ms-transform: scaleX(.8); -o-transform: scaleX(.8); transform: scaleX(.8)}
.MJXp-scale9 {-webkit-transform: scaleX(.9); -moz-transform: scaleX(.9); -ms-transform: scaleX(.9); -o-transform: scaleX(.9); transform: scaleX(.9)}
.MathJax_PHTML .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}
</style></head>

<body class="single single-post postid-155868 single-format-standard wpusb-position-fixed-active"><div id="MathJax_Message" style="display: none;"></div>
<!-- begin header -->
<div id="header">
		    	    	<a href="https://www.r-bloggers.com/" class="headerimage"><img src="./Genome-wide association studies in R _ R-bloggers_files/R_02_2016-05-01.png" alt="R-bloggers" class="aligncenter" scale="0"></a>
        <div id="description" class="aligncenter">
        R news and tutorials contributed by (750) R bloggers    </div>     
    <div class="main-navigation"><ul id="menu-top-nav" class="sf-menu sf-js-enabled sf-shadow"><li id="menu-item-48314" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-home menu-item-48314"><a href="https://www.r-bloggers.com/">Home</a></li>
<li id="menu-item-92333" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-92333"><a href="https://www.r-bloggers.com/about/">About</a></li>
<li id="menu-item-50111" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-50111"><a href="https://feeds.feedburner.com/RBloggers">RSS</a></li>
<li id="menu-item-48313" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-48313"><a href="https://www.r-bloggers.com/add-your-blog/">add your blog!</a></li>
<li id="menu-item-111419" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-111419"><a href="https://www.r-bloggers.com/how-to-learn-r-2/">Learn R</a></li>
<li id="menu-item-75513" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-has-children menu-item-75513"><a href="https://www.r-users.com/" class="sf-with-ul">R jobs<span class="sf-sub-indicator">�</span><span class="sf-sub-indicator">�</span><span class="sf-sub-indicator">�</span></a>
<ul class="sub-menu" style="float: none; width: 12em; display: none; visibility: hidden;">
	<li id="menu-item-76945" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-76945" style="white-space: normal; float: left; width: 100%;"><a href="https://www.r-users.com/submit-job/" style="float: none; width: auto;">Submit a new job (it’s free)</a></li>
	<li id="menu-item-76946" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-76946" style="white-space: normal; float: left; width: 100%;"><a href="https://www.r-users.com/" style="float: none; width: auto;">Browse latest jobs (also free)</a></li>
</ul>
</li>
<li id="menu-item-48311" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-48311"><a href="https://www.r-bloggers.com/contact-us/">Contact us</a></li>
</ul></div>
</div>
<!-- end header -->


<div id="mainwrapper">
<!-- begin sidebar -->
    <div id="sidebar">
		<div class="side-widget"><h2>Welcome!</h2>			<div class="textwidget"><iframe allowtransparency="true" frameborder="0" scrolling="no" src="./Genome-wide association studies in R _ R-bloggers_files/follow_button.html" style="width:164px; height:30px;"></iframe>


Here you will find daily <strong>news and tutorials about R</strong>, contributed by over 750 bloggers. 
<br>
There are many ways to <strong>follow us - </strong>
<br>
<a href="https://feedburner.google.com/fb/a/mailverify?uri=RBloggers">By e-mail:</a>

<form style="border:1px solid #ccc;padding:3px;text-align:center; background: none repeat scroll 0 0 #FDEADA;" action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#39;https://feedburner.google.com/fb/a/mailverify?uri=RBloggers&#39;, &#39;popupwindow&#39;, &#39;scrollbars=yes,width=550,height=520&#39;);return true">
<input type="text" style="width:110px" onclick="if (this.value == &#39;Your e-mail here&#39;) this.value = &#39;&#39;;" value="Your e-mail here" name="email">
<input type="hidden" value="RBloggers" name="uri"><input type="hidden" name="loc" value="en_US"><input type="submit" value="Subscribe">
<a href="https://feeds.feedburner.com/RBloggers"><img src="./Genome-wide association studies in R _ R-bloggers_files/RBloggers" height="26" width="88" style="border:0" alt="" scale="0"></a>
</form>



<a href="https://www.facebook.com/rbloggers">On Facebook:</a>
<iframe src="./Genome-wide association studies in R _ R-bloggers_files/likebox.html" scrolling="no" frameborder="0" style="border:none; overflow:hidden; width:160px; height:238px;" allowtransparency="true"></iframe>


<br>
<strong>If you are an R blogger yourself</strong> you are invited to <a href="https://www.r-bloggers.com/add-your-blog/">add your own R content feed to this site</a> (<strong>Non-English</strong> R bloggers should add themselves- <a href="https://www.r-bloggers.com/lang/add-your-blog">here</a>)</div>
		</div><div class="side-widget"><h2><a class="rsswidget" href="https://feeds.feedburner.com/Rjobs"><img class="rss-widget-icon" style="border:0" width="14" height="14" src="./Genome-wide association studies in R _ R-bloggers_files/rss.png" alt="RSS" scale="0"></a> <a class="rsswidget" href="https://www.r-users.com/">Jobs for R-users</a></h2><ul><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/OMCwwBo1veo/">Quantitative Econometrician @ San Francisco, California, U.S.</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/X3DpiECQ8DE/">Postdoctoral Research Fellow in Healthcare Systems Engineering @ Maryland, U.S.</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/xPASaTEn8kE/">R Programmer &amp; Statistician for Academic Research</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/8QL91gt7zzc/">Data Scientist</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/oyehz_z-zrw/">Postdoctoral Data Scientist: GIS and mHealth</a></li></ul></div><div class="side-widget"><h2>Popular Searches</h2></div>		<div class="side-widget">		<h2>Recent Posts</h2>		<ul>
					<li>
				<a href="https://www.r-bloggers.com/genome-wide-association-studies-in-r/">Genome-wide association studies in R</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/rstudio-v1-1-released/">RStudio v1.1 Released</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/celebrating-20-years-of-cran/">Celebrating 20 years of CRAN</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/summer-astrostat-school-room-with-a-view-jatp/">[summer Astrostat school] room with a view [jatp]</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/how-to-sample-from-multidimensional-distributions-using-gibbs-sampling/">How to sample from multidimensional distributions using Gibbs sampling?</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/mea-culpa/">mea culpa!</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/a-step-change-in-managing-your-calendar-without-social-media/">A step change in managing your calendar, without social media</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/spatial-networks-case-study-st-james-centre-edinburgh-13/">Spatial networks – case study St James centre, Edinburgh (1/3)</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/its-tibbletime-v0-0-2-time-aware-tibbles-new-functions-weather-analysis-and-more/">It’s tibbletime v0.0.2: Time-Aware Tibbles, New Functions, Weather Analysis and More</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/nfl-series/">NFL Series</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/linear-logistic-regression-in-r-dealing-with-unknown-factor-levels-in-test-data/">Linear / Logistic Regression in R: Dealing With Unknown Factor Levels in Test Data</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/data-visualization-course-for-first-year-students/">Data Visualization Course for First-Year Students</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/arima-models-and-intervention-analysis/">ARIMA models and Intervention Analysis</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/from-static-to-numeric-to-single-choice-exercises-in-rexams/">From Static to Numeric to Single-Choice Exercises in R/exams</a>
						</li>
					<li>
				<a href="https://www.r-bloggers.com/getsymbols-and-alpha-vantage/">getSymbols and Alpha Vantage</a>
						</li>
				</ul>
		</div>		<div class="side-widget"><h2>Other sites</h2>
	<ul class="xoxo blogroll">
<li><a href="http://www.proc-x.com/" title="SAS news gathered from bloggers">SAS blogs</a></li>
<li><a href="https://www.r-users.com/">Jobs for R-users</a></li>

	</ul>
</div>
    </div>
<!-- end sidebar -->	<div id="leftcontent">
	
		

	<div id="post-155868" class="post-155868 post type-post status-publish format-standard hentry category-r-bloggers">
					
				
                <h1>Genome-wide association studies in R</h1>
                <div class="meta"><div class="date">October 9, 2017</div>By <a href="https://www.r-bloggers.com/author/francisco-lima/" title="Posts by Francisco Lima" rel="author">Francisco Lima</a></div><div class="entry"><p class="syndicated-attribution"></p><div class="social4i" style="height:29px;"><div class="social4in" style="height:29px;float: left;"><div class="socialicons s4fblike" style="float:left;margin-right: 10px;"><div class="fb-like fb_iframe_widget" data-href="https://www.r-bloggers.com/genome-wide-association-studies-in-r/" data-send="true" data-layout="button_count" data-width="100" data-height="21" data-show-faces="false" fb-xfbml-state="rendered" fb-iframe-plugin-query="app_id=&amp;container_width=0&amp;height=21&amp;href=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F&amp;layout=button_count&amp;locale=en_US&amp;sdk=joey&amp;send=true&amp;show_faces=false&amp;width=100"><span style="vertical-align: bottom; width: 118px; height: 20px;"><iframe name="f313ed70642328" width="100px" height="21px" frameborder="0" allowtransparency="true" allowfullscreen="true" scrolling="no" title="fb:like Facebook Social Plugin" src="./Genome-wide association studies in R _ R-bloggers_files/like.html" style="border: none; visibility: visible; width: 118px; height: 20px;" class=""></iframe></span></div></div><div class="socialicons s4linkedin" style="float:left;margin-right: 10px;"><span class="IN-widget" style="line-height: 1; vertical-align: baseline; display: inline-block; text-align: center;"><span style="padding: 0px !important; margin: 0px !important; text-indent: 0px !important; display: inline-block !important; vertical-align: baseline !important; font-size: 1px !important;"><span id="li_ui_li_gen_1507763900872_0"><a id="li_ui_li_gen_1507763900872_0-link" href="javascript:void(0);"><span id="li_ui_li_gen_1507763900872_0-logo">in</span><span id="li_ui_li_gen_1507763900872_0-title"><span id="li_ui_li_gen_1507763900872_0-mark"></span><span id="li_ui_li_gen_1507763900872_0-title-text">Share</span></span></a></span></span><span style="padding: 0px !important; margin: 0px !important; text-indent: 0px !important; display: inline-block !important; vertical-align: baseline !important; font-size: 1px !important;"><span id="li_ui_li_gen_1507763900889_1-container" class="IN-right"><span id="li_ui_li_gen_1507763900889_1" class="IN-right"><span id="li_ui_li_gen_1507763900889_1-inner" class="IN-right"><span id="li_ui_li_gen_1507763900889_1-content" class="IN-right">10</span></span></span></span></span></span><script type="in/share+init" data-url="https://www.r-bloggers.com/genome-wide-association-studies-in-r/" data-counter="right"></script></div></div><div style="clear:both"></div></div>

<div style="border: 1px solid; background: none repeat scroll 0 0 #EDEDED; margin: 1px; font-size: 12px;">
(This article was first published on  <strong><a href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/"> poissonisfish</a></strong>, and kindly contributed to <a href="https://www.r-bloggers.com/" rel="nofollow">R-bloggers)</a>      
</div><p></p>
<aside class="mashsb-container mashsb-main mashsb-stretched"><div class="mashsb-box"><div class="mashsb-count mash-small" style="float:left;"><div class="counts mashsbcount">20</div><span class="mashsb-sharetext">SHARES</span></div><div class="mashsb-buttons"><a class="mashicon-facebook mash-small mash-center mashsb-noshadow" href="https://www.facebook.com/sharer.php?u=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F" target="_blank" rel="nofollow"><span class="icon"></span><span class="text">Share</span></a><a class="mashicon-twitter mash-small mash-center mashsb-noshadow" href="https://twitter.com/intent/tweet?text=Genome-wide%20association%20studies%20in%20R&amp;url=https://www.r-bloggers.com/genome-wide-association-studies-in-r/&amp;via=Rbloggers" target="_blank" rel="nofollow"><span class="icon"></span><span class="text">Tweet</span></a><div class="onoffswitch2 mash-small mashsb-noshadow" style="display:none;"></div></div>
            </div>
                <div style="clear:both;"></div></aside>
            <!-- Share buttons by mashshare.net - Version: 3.4.4--><figure data-shortcode="caption" id="attachment_4185" style="width: 1920px" class="wp-caption alignnone"><img src="./Genome-wide association studies in R _ R-bloggers_files/maxresdefault.jpg" data-attachment-id="4185" data-permalink="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/maxresdefault/" data-orig-file="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456" data-orig-size="1920,1080" data-comments-opened="1" data-image-title="maxresdefault" data-image-description="" data-medium-file="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456?w=300" data-large-file="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456?w=1024" class="alignnone size-full wp-image-4185" alt="maxresdefault" sizes="(max-width: 1108px) 100vw, 1108px" data-lazy-loaded="true" style="display: inline;" width="444" height="250" originalw="456" src-orig="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456" scale="1.375"><noscript>&lt;img data-attachment-id="4185" data-permalink="<a class="vglnk" href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/maxresdefault/" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>09</span><span>/</span><span>genome</span><span>-</span><span>wide</span><span>-</span><span>association</span><span>-</span><span>studies</span><span>-</span><span>in</span><span>-</span><span>r</span><span>/</span><span>maxresdefault</span><span>/</span></a>" data-orig-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" data-orig-size="1920,1080" data-comments-opened="1" data-image-meta="{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}" data-image-title="maxresdefault" data-image-description="" data-medium-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>300</span></a>" data-large-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456?w=1024" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>1024</span></a>" class="alignnone size-full wp-image-4185" src="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" alt="maxresdefault" srcset_temp="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>456</span></a> 1108w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=150" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>150</span></a> 150w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>300</span></a> 300w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=768" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>768</span></a> 768w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg?w=1024" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span><span>?</span><span>w</span><span>=</span><span>1024</span></a> 1024w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/maxresdefault.jpg" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>maxresdefault</span><span>.</span><span>jpg</span></a> 1920w" sizes="(max-width: 1108px) 100vw, 1108px"   data-recalc-dims="1" /&gt;</noscript><br>
<figcaption class="wp-caption-text">’10 High Cholesterol Foods You Must Avoid’. From:&nbsp;<a class="vglnk" href="https://www.youtube.com/watch?v=RKZ1Wk8tIZo" rel="nofollow"><span>https</span><span>://</span><span>www</span><span>.</span><span>youtube</span><span>.</span><span>com</span><span>/</span><span>watch</span><span>?</span><span>v</span><span>=</span><span>RKZ1Wk8tIZo</span></a></figcaption>
</figure>
<p>This time I elaborate on a much more specific subject that will mostly concern biologists and geneticists. I will try my best to outline the approach as to ensure non-experts will still have a basic understanding. This tutorial illustrates the power of genome-wide association (GWA) studies by mapping the genetic determinants of cholesterol levels using three Southeast Asian populations.</p>
<h2>Historical background</h2>
<p>Early since Charles Darwin formulated the theories of natural and sexual selection in the late 1800s, the underlying role of genes, each represented by different alleles (i.e. variants) in different individuals was yet to be elucidated. His younger contemporary fellow Gregor Mendel scratched the surface after identifying the mechanism of trait inheritance, genetic segregation. In the years that followed, the genetic basis of phenotypes (i.e. observable traits) was gradually unraveled by classic geneticists pioneered by Ronald Fisher, who introduced key concepts such as genetic variance. The then emerging concept of genotype (i.e. genetic make-up) required the development of polymorphic genetic markers. In the early days, genotyping was based on determining the allelic composition of loci (loose definition of chromosomal regions), and later of copy number variations (CNVs), short-tandem repeats (STRs) and single nucleotide polymorphisms (SNPs). Humans and many other organisms including plants are diploid (i.e. carry two copies of each chromosome), which implies bi-allelic markers with alleles e.g.&nbsp;<em>A</em> and <em>a</em> distinguish individuals into <em>AA</em> (homozygous dominant), <em>Aa</em> (heterozygous) and <em>aa</em> (homozygous recessive).</p>
<p>The early discovery of the genetic basis for disorders such as sickle cell anemia and haemophilia owes much to their relatively simple genetic architectures – few mutations with high penetrance, thus more easily identified. Understandably then, more complex polygenic diseases&nbsp;that have been around for a long time, such as the neurodegenerative Alzheimer and Parkinson, are still currently under investigation. To characterize such traits, products of the interplay of many genes with small effects, it takes large genetic resources as well as flexible statistical methods, which is no problem in the current era of omics data and high-performance computing.</p>
<h2>Genome-wide association studies</h2>
<p>Genome-wide association (GWA) studies scan an entire species genome for association between up to millions of SNPs and a given trait of interest. Notably, the trait of interest can be virtually any sort of phenotype ascribed to the population, be it qualitative (e.g. disease status) or quantitative (e.g. height). Essentially, given <em>p</em> SNPs and <em>n</em> samples or individuals, a GWA analysis will fit&nbsp;<em>p </em>independent<em>&nbsp;</em>univariate<em>&nbsp;</em>linear models, each based on <em>n</em> samples, using the genotype of each SNP as predictor of the trait of interest. The significance of association (<em>P</em>-value) in each of the&nbsp;<em>p</em> tests is determined from the&nbsp;coefficient estimate <img src="./Genome-wide association studies in R _ R-bloggers_files/latex.php" alt="\beta " title="\beta " class="latex" data-lazy-loaded="true" style="display: inline;" width="10" height="14" src-orig="https://s0.wp.com/latex.php?latex=%5Cbeta+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=%5Cbeta+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=%</span><span>5Cbeta</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="\beta " title="\beta " class="latex" /&gt;</noscript> of the corresponding SNP (technically speaking, the significance of association is <img src="./Genome-wide association studies in R _ R-bloggers_files/latex(1).php" alt="P(\beta|H_0: \beta=0) " title="P(\beta|H_0: \beta=0) " class="latex" data-lazy-loaded="true" style="display: inline;" width="112" height="18" src-orig="https://s0.wp.com/latex.php?latex=P%28%5Cbeta%7CH_0%3A+%5Cbeta%3D0%29+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=P%28%5Cbeta%7CH_0%3A+%5Cbeta%3D0%29+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>P</span><span>%</span><span>28</span><span>%</span><span>5Cbeta</span><span>%</span><span>7CH</span><span>_</span><span>0</span><span>%</span><span>3A</span><span>+%</span><span>5Cbeta</span><span>%</span><span>3D0</span><span>%</span><span>29</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="P(\beta|H_0: \beta=0) " title="P(\beta|H_0: \beta=0) " class="latex" /&gt;</noscript>). Note that because these tests are independent and quite numerous, there is a great computational advantage in setting up a parallelized GWA analysis (as we will do shortly). Quite reasonably, it is necessary to adjust the resulting <em>P-</em>values using multiple hypothesis testing methods such as Bonferroni, Benjamini-Hochberg or false discovery rate (FDR).&nbsp;GWA studies are now commonplace in genetics of many different species.</p>
<h2>Association mapping vs. linkage mapping</h2>
<p>Too often, people cannot tell the difference between association and linkage mapping, or quantitative trait loci (QTL) mapping. Albeit conceptually similar, their are actually opposite in their workings. One of the key differences between the two is that association mapping relies on high-density SNP genotyping of unrelated individuals, whereas linkage mapping relies on the segregation of substantially fewer markers in controlled breeding experiments – unsurprisingly QTL mapping is seldom conducted in humans. Importantly, association mapping gives you point associations in the genome, whereas linkage mapping gives you QTL, chromosomal regions.</p>
<p>The present tutorial covers fundamental aspects to consider when conducting GWA analysis, from the pre-processing of genotype and phenotype data to the interpretation of results. We will use a mixed population of 316 Chinese, Indian and Malay that was recently characterized using high-throughput SNP-chip sequencing, transcriptomics and lipidomics (<a href="https://www.nature.com/articles/s41467-017-00413-x" rel="nofollow" target="_blank">Saw et al., 2017</a>). More specifically, we will search for associations between the &gt;2.5 million SNP markers and cholesterol levels. Finally, we will explore the vicinity of candidate SNPs using the USCS Genome Browser in order to gain functional insights. The methodology shown here is largely based on the tutorial outlined in&nbsp;<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6605/full" rel="nofollow" target="_blank">Reed et al., 2015</a>. The R scripts and some of the data can be found in my <a href="https://github.com/monogenea/GWAStutorial" rel="nofollow" target="_blank">repository</a>, but you will still need to download the omics data from&nbsp;<a href="http://phg.nus.edu.sg/StatGen/public_html/Iomics/downloads/iOmics_data.tar.gz" rel="nofollow" target="_blank">here</a>. Please follow the instructions in the repo.</p>
<h1>Let’s get started with R</h1>
<h2>Read data</h2>
<p>Let’s first import the PLINK-converted .bed, .fam and .bim Illumina files from each of the three ethnic groups. We will use the function <strong>read.plink</strong> from the package <strong>snpStats</strong>&nbsp;and work on the resulting objects throughout the rest of the tutorial. This function reads .bed, .fam and .bim and creates a list of three elements – <strong>$genotypes</strong>, <strong>$fam</strong> and <strong>$map</strong>. The first&nbsp;contains all SNPs determined from all samples, the second contains information about pedigree and sex, and the third&nbsp;contains the genomic coordinates of the SNPs. At this point we have a total of 323 individuals (110 Chinese, 105 Indian and 108 Malay) and&nbsp;2,527,458 SNPs.&nbsp;Next, we will change the Illumina SNP IDs stored in <strong>$genotype</strong> to the more conventional rs IDs, which will allow us to zoom in into the genomic regions that surround the candidate SNPs in the USCS Genome Browser. I prepared a table that establishes the correspondence between the two, so that we can easily switch the IDs.</p>
<div><div id="highlighter_350886" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div><div class="line number12 index11 alt1">12</div><div class="line number13 index12 alt2">13</div><div class="line number14 index13 alt1">14</div><div class="line number15 index14 alt2">15</div><div class="line number16 index15 alt1">16</div><div class="line number17 index16 alt2">17</div><div class="line number18 index17 alt1">18</div><div class="line number19 index18 alt2">19</div><div class="line number20 index19 alt1">20</div><div class="line number21 index20 alt2">21</div><div class="line number22 index21 alt1">22</div><div class="line number23 index22 alt2">23</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r functions">library</code><code class="r plain">(snpStats)</code></div><div class="line number2 index1 alt1"><code class="r functions">load</code><code class="r plain">(</code><code class="r string">"conversionTable.RData"</code><code class="r plain">)</code></div><div class="line number3 index2 alt2">&nbsp;</div><div class="line number4 index3 alt1"><code class="r plain">pathM &lt;- </code><code class="r functions">paste</code><code class="r plain">(</code><code class="r string">"Genomics/108Malay_2527458snps"</code><code class="r plain">, </code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">".bed"</code><code class="r plain">, </code><code class="r string">".bim"</code><code class="r plain">, </code><code class="r string">".fam"</code><code class="r plain">), sep = </code><code class="r string">""</code><code class="r plain">)</code></div><div class="line number5 index4 alt2"><code class="r plain">SNP_M &lt;- </code><code class="r functions">read.plink</code><code class="r plain">(pathM[1], pathM[2], pathM[3])</code></div><div class="line number6 index5 alt1">&nbsp;</div><div class="line number7 index6 alt2"><code class="r plain">pathI &lt;- </code><code class="r functions">paste</code><code class="r plain">(</code><code class="r string">"Genomics/105Indian_2527458snps"</code><code class="r plain">, </code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">".bed"</code><code class="r plain">, </code><code class="r string">".bim"</code><code class="r plain">, </code><code class="r string">".fam"</code><code class="r plain">), sep = </code><code class="r string">""</code><code class="r plain">)</code></div><div class="line number8 index7 alt1"><code class="r plain">SNP_I &lt;- </code><code class="r functions">read.plink</code><code class="r plain">(pathI[1], pathI[2], pathI[3])</code></div><div class="line number9 index8 alt2">&nbsp;</div><div class="line number10 index9 alt1"><code class="r plain">pathC &lt;- </code><code class="r functions">paste</code><code class="r plain">(</code><code class="r string">"Genomics/110Chinese_2527458snps"</code><code class="r plain">, </code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">".bed"</code><code class="r plain">, </code><code class="r string">".bim"</code><code class="r plain">, </code><code class="r string">".fam"</code><code class="r plain">), sep = </code><code class="r string">""</code><code class="r plain">)</code></div><div class="line number11 index10 alt2"><code class="r plain">SNP_C &lt;- </code><code class="r functions">read.plink</code><code class="r plain">(pathC[1], pathC[2], pathC[3])</code></div><div class="line number12 index11 alt1">&nbsp;</div><div class="line number13 index12 alt2"><code class="r comments"># Merge the three SNP datasets</code></div><div class="line number14 index13 alt1"><code class="r plain">SNP &lt;- </code><code class="r functions">rbind</code><code class="r plain">(SNP_M$genotypes, SNP_I$genotypes, SNP_C$genotypes)</code></div><div class="line number15 index14 alt2">&nbsp;</div><div class="line number16 index15 alt1"><code class="r comments"># Take one bim map (all 3 maps are based on the same ordered set of SNPs)</code></div><div class="line number17 index16 alt2"><code class="r plain">map &lt;- SNP_M$map</code></div><div class="line number18 index17 alt1"><code class="r functions">colnames</code><code class="r plain">(map) &lt;- </code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">"chr"</code><code class="r plain">, </code><code class="r string">"SNP"</code><code class="r plain">, </code><code class="r string">"gen.dist"</code><code class="r plain">, </code><code class="r string">"position"</code><code class="r plain">, </code><code class="r string">"A1"</code><code class="r plain">, </code><code class="r string">"A2"</code><code class="r plain">)</code></div><div class="line number19 index18 alt2">&nbsp;</div><div class="line number20 index19 alt1"><code class="r comments"># Rename SNPs present in the conversion table into rs IDs</code></div><div class="line number21 index20 alt2"><code class="r plain">mappedSNPs &lt;- </code><code class="r functions">intersect</code><code class="r plain">(map$SNP, </code><code class="r functions">names</code><code class="r plain">(conversionTable))</code></div><div class="line number22 index21 alt1"><code class="r plain">newIDs &lt;- conversionTable[</code><code class="r functions">match</code><code class="r plain">(map$SNP[map$SNP %</code><code class="r keyword">in</code><code class="r plain">% mappedSNPs], </code><code class="r functions">names</code><code class="r plain">(conversionTable))]</code></div><div class="line number23 index22 alt2"><code class="r plain">map$SNP[</code><code class="r functions">rownames</code><code class="r plain">(map) %</code><code class="r keyword">in</code><code class="r plain">% mappedSNPs] &lt;- newIDs</code></div></div></td></tr></tbody></table></div></div>
<p>Next we import and merge the three lipid data sets (stored as .txt) and determine which samples are present in both SNP and lipid data sets. In the following analyses we will use the subset of samples profiled in both platforms, a total of 319. Finally, we create a list <strong>genData</strong> that stores the merged SNP data (<strong>$SNP</strong>), one of the three <strong>$map</strong> since they are all identical (<strong>$MAP</strong>) and the merged lipid data (<strong>$LIP</strong>). Finally, let’s save RAM for the subsequent steps and remove all files after saving <strong>genData</strong> into a <strong>.RData</strong> file.</p>
<div><div id="highlighter_733802" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div><div class="line number12 index11 alt1">12</div><div class="line number13 index12 alt2">13</div><div class="line number14 index13 alt1">14</div><div class="line number15 index14 alt2">15</div><div class="line number16 index15 alt1">16</div><div class="line number17 index16 alt2">17</div><div class="line number18 index17 alt1">18</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># Load lipid datasets &amp; match SNP-Lipidomics samples</code></div><div class="line number2 index1 alt1"><code class="r plain">lipidsMalay &lt;- </code><code class="r functions">read.delim</code><code class="r plain">(</code><code class="r string">"Lipidomic/117Malay_282lipids.txt"</code><code class="r plain">, row.names = 1)</code></div><div class="line number3 index2 alt2"><code class="r plain">lipidsIndian &lt;- </code><code class="r functions">read.delim</code><code class="r plain">(</code><code class="r string">"Lipidomic/120Indian_282lipids.txt"</code><code class="r plain">, row.names = 1)</code></div><div class="line number4 index3 alt1"><code class="r plain">lipidsChinese &lt;- </code><code class="r functions">read.delim</code><code class="r plain">(</code><code class="r string">"Lipidomic/122Chinese_282lipids.txt"</code><code class="r plain">, row.names = 1)</code></div><div class="line number5 index4 alt2"><code class="r functions">all</code><code class="r plain">(</code><code class="r functions">Reduce</code><code class="r plain">(intersect, </code><code class="r functions">list</code><code class="r plain">(</code><code class="r functions">colnames</code><code class="r plain">(lipidsMalay),</code></div><div class="line number6 index5 alt1"><code class="r functions">colnames</code><code class="r plain">(lipidsIndian),</code></div><div class="line number7 index6 alt2"><code class="r functions">colnames</code><code class="r plain">(lipidsChinese))) == </code><code class="r functions">colnames</code><code class="r plain">(lipidsMalay)) </code><code class="r comments"># TRUE</code></div><div class="line number8 index7 alt1"><code class="r plain">lip &lt;- </code><code class="r functions">rbind</code><code class="r plain">(lipidsMalay, lipidsIndian, lipidsChinese)</code></div><div class="line number9 index8 alt2">&nbsp;</div><div class="line number10 index9 alt1"><code class="r plain">matchingSamples &lt;- </code><code class="r functions">intersect</code><code class="r plain">(</code><code class="r functions">rownames</code><code class="r plain">(lip), </code><code class="r functions">rownames</code><code class="r plain">(SNP))</code></div><div class="line number11 index10 alt2"><code class="r plain">SNP &lt;- SNP[matchingSamples,]</code></div><div class="line number12 index11 alt1"><code class="r plain">lip &lt;- lip[matchingSamples,]</code></div><div class="line number13 index12 alt2">&nbsp;</div><div class="line number14 index13 alt1"><code class="r plain">genData &lt;- </code><code class="r functions">list</code><code class="r plain">(SNP = SNP, MAP = map, LIP = lip)</code></div><div class="line number15 index14 alt2"><code class="r functions">save</code><code class="r plain">(genData, file = </code><code class="r string">"PhenoGenoMap.RData"</code><code class="r plain">)</code></div><div class="line number16 index15 alt1">&nbsp;</div><div class="line number17 index16 alt2"><code class="r comments"># Clear memory</code></div><div class="line number18 index17 alt1"><code class="r functions">rm</code><code class="r plain">(list = </code><code class="r functions">ls</code><code class="r plain">())</code></div></div></td></tr></tbody></table></div></div>
<h2>Pre-processing</h2>
<p>Let’s reload <strong>genData</strong> and clean it up. In a nutshell, the pre-processing of the data consists in</p>
<ul>
<li>discarding SNPs with call rate<strong>&nbsp;</strong>&lt; 1 &nbsp;or &nbsp;MAF &lt; 0.1</li>
<li>discarding samples with call rate &lt; 100%, IBD kinship coefficient &gt; 0.1 or inbreeding coefficient |<em>F</em>| &gt; 0.1</li>
</ul>
<p>Call rate is the proportion of SNPs (or samples) that were genotyped. For example, a call rate of 0.95 for a particular SNP (sample) means 5% of the values are missing. Because we have so many SNPs, we can afford to have absolutely no missing values in the <strong>$SNP</strong> matrix by imposing a call rate threshold of 1 for both SNPs and samples. If you want to relax the threshold and tolerate missing values, bear in mind you need to run a whole procedure for imputing those. <a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6605/full" rel="nofollow" target="_blank">Reed et al., 2015</a>&nbsp;describe a PCA-based imputation method that utilizes the 1,000 Genome Project as a proxy, in case you are interested.</p>
<p>Minor-allele frequency (MAF) denotes the proportion of the least common allele for each SNP. Of course, it is harder to detect associations with rare variants and this is why we select against low MAF values. Most GWA studies I have read typically report MAF thresholds of 0.05. Here, I opt for a more stringent 0.1 because again, we have plenty of data and since all this is for illustrative purposes we want to conduct a fast GWA analysis.</p>
<div><div id="highlighter_523818" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div><div class="line number12 index11 alt1">12</div><div class="line number13 index12 alt2">13</div><div class="line number14 index13 alt1">14</div><div class="line number15 index14 alt2">15</div><div class="line number16 index15 alt1">16</div><div class="line number17 index16 alt2">17</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r functions">library</code><code class="r plain">(snpStats)</code></div><div class="line number2 index1 alt1"><code class="r functions">library</code><code class="r plain">(doParallel)</code></div><div class="line number3 index2 alt2"><code class="r functions">library</code><code class="r plain">(SNPRelate)</code></div><div class="line number4 index3 alt1"><code class="r functions">library</code><code class="r plain">(GenABEL)</code></div><div class="line number5 index4 alt2"><code class="r functions">library</code><code class="r plain">(dplyr)</code></div><div class="line number6 index5 alt1"><code class="r functions">source</code><code class="r plain">(</code><code class="r string">"GWASfunction.R"</code><code class="r plain">)</code></div><div class="line number7 index6 alt2"><code class="r functions">load</code><code class="r plain">(</code><code class="r string">"PhenoGenoMap.RData"</code><code class="r plain">)</code></div><div class="line number8 index7 alt1">&nbsp;</div><div class="line number9 index8 alt2"><code class="r comments"># Use SNP call rate of 100%, MAF of 0.1 (very stringent)</code></div><div class="line number10 index9 alt1"><code class="r plain">maf &lt;- 0.1</code></div><div class="line number11 index10 alt2"><code class="r plain">callRate &lt;- 1</code></div><div class="line number12 index11 alt1"><code class="r plain">SNPstats &lt;- </code><code class="r functions">col.summary</code><code class="r plain">(genData$SNP)</code></div><div class="line number13 index12 alt2">&nbsp;</div><div class="line number14 index13 alt1"><code class="r plain">maf_call &lt;- </code><code class="r functions">with</code><code class="r plain">(SNPstats, MAF &gt; maf &amp; Call.rate == callRate)</code></div><div class="line number15 index14 alt2"><code class="r plain">genData$SNP &lt;- genData$SNP[,maf_call]</code></div><div class="line number16 index15 alt1"><code class="r plain">genData$MAP &lt;- genData$MAP[maf_call,]</code></div><div class="line number17 index16 alt2"><code class="r plain">SNPstats &lt;- SNPstats[maf_call,]</code></div></div></td></tr></tbody></table></div></div>
<p>Next, we need to consider samples that exhibit excessive heterozygosity – technically speaking, deviations from the Hardy-Weinberg equilibrium (HWE),</p>
<p><img src="./Genome-wide association studies in R _ R-bloggers_files/latex(2).php" alt="p^2 + 2pq + q^2 = 1 " title="p^2 + 2pq + q^2 = 1 " class="latex" data-lazy-loaded="true" style="display: inline;" width="123" height="17" src-orig="https://s0.wp.com/latex.php?latex=p%5E2+%2B+2pq+%2B+q%5E2+%3D+1+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=2" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=p%5E2+%2B+2pq+%2B+q%5E2+%3D+1+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=2" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>p</span><span>%</span><span>5E2</span><span>+%</span><span>2B</span><span>+</span><span>2pq</span><span>+%</span><span>2B</span><span>+</span><span>q</span><span>%</span><span>5E2</span><span>+%</span><span>3D</span><span>+</span><span>1</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>2</span></a>" alt="p^2 + 2pq + q^2 = 1 " title="p^2 + 2pq + q^2 = 1 " class="latex" /&gt;</noscript></p>
<p>where <img src="./Genome-wide association studies in R _ R-bloggers_files/latex(3).php" alt="p " title="p " class="latex" data-lazy-loaded="true" style="display: inline;" width="9" height="10" src-orig="https://s0.wp.com/latex.php?latex=p+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=p+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>p</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="p " title="p " class="latex" /&gt;</noscript>&nbsp;and <img src="./Genome-wide association studies in R _ R-bloggers_files/latex(4).php" alt="q " title="q " class="latex" data-lazy-loaded="true" style="display: inline;" width="8" height="10" src-orig="https://s0.wp.com/latex.php?latex=q+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=q+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>q</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="q " title="q " class="latex" /&gt;</noscript>&nbsp;are the frequencies of two alleles <em>A</em> and <em>a</em> that sum up to one, and <img src="./Genome-wide association studies in R _ R-bloggers_files/latex(5).php" alt="2pq " title="2pq " class="latex" data-lazy-loaded="true" style="display: inline;" width="24" height="15" src-orig="https://s0.wp.com/latex.php?latex=2pq+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=2pq+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>2pq</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="2pq " title="2pq " class="latex" /&gt;</noscript> the frequency of heterozygous individuals, assuming bi-allelic SNPs.&nbsp;As such, we determine the inbreeding coefficient |<em>F|,</em></p>
<p><img src="./Genome-wide association studies in R _ R-bloggers_files/latex(6).php" alt="|F| = 1 - \frac{H}{2pq} " title="|F| = 1 - \frac{H}{2pq} " class="latex" data-lazy-loaded="true" style="display: inline;" width="91" height="22" src-orig="https://s0.wp.com/latex.php?latex=%7CF%7C+%3D+1+-+%5Cfrac%7BH%7D%7B2pq%7D+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=2" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=%7CF%7C+%3D+1+-+%5Cfrac%7BH%7D%7B2pq%7D+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=2" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=%</span><span>7CF</span><span>%</span><span>7C</span><span>+%</span><span>3D</span><span>+</span><span>1</span><span>+-+%</span><span>5Cfrac</span><span>%</span><span>7BH</span><span>%</span><span>7D</span><span>%</span><span>7B2pq</span><span>%</span><span>7D</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>2</span></a>" alt="|F| = 1 - \frac{H}{2pq} " title="|F| = 1 - \frac{H}{2pq} " class="latex" /&gt;</noscript></p>
<p>where <em>H&nbsp;</em>and&nbsp;<img src="./Genome-wide association studies in R _ R-bloggers_files/latex(5).php" alt="2pq " title="2pq " class="latex" data-lazy-loaded="true" style="display: inline;" width="24" height="15" src-orig="https://s0.wp.com/latex.php?latex=2pq+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=2pq+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>2pq</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="2pq " title="2pq " class="latex" /&gt;</noscript> are the observed and expected heterozygosity (i.e. proportion of heterozygous SNPs per sample), respectively. You will note that in practice, rather than using the proportions we will use counts which still gives us the same results. Overall, large and small |<em>F</em>| might indicate poor sample quality or&nbsp;inbreeding, respectively. We will exclude samples with |<em>F</em>|&gt; 0.1 as described in&nbsp;<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6605/full" rel="nofollow" target="_blank">Reed et al., 2015</a>.</p>
<div><div id="highlighter_4494" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># Sample call rate &amp; heterozygosity</code></div><div class="line number2 index1 alt1"><code class="r plain">callMat &lt;- !</code><code class="r functions">is.na</code><code class="r plain">(genData$SNP)</code></div><div class="line number3 index2 alt2"><code class="r plain">Sampstats &lt;- </code><code class="r functions">row.summary</code><code class="r plain">(genData$SNP)</code></div><div class="line number4 index3 alt1"><code class="r plain">hetExp &lt;- callMat %*% (2 * SNPstats$MAF * (1 - SNPstats$MAF)) </code><code class="r comments"># Hardy-Weinberg heterozygosity (expected)</code></div><div class="line number5 index4 alt2"><code class="r plain">hetObs &lt;- </code><code class="r functions">with</code><code class="r plain">(Sampstats, Heterozygosity * (</code><code class="r functions">ncol</code><code class="r plain">(genData$SNP)) * Call.rate)</code></div><div class="line number6 index5 alt1"><code class="r plain">Sampstats$hetF &lt;- 1-(hetObs/hetExp)</code></div><div class="line number7 index6 alt2"><code class="r comments"># Use sample call rate of 100%, het threshold of 0.1 (very stringent)</code></div><div class="line number8 index7 alt1"><code class="r plain">het &lt;- 0.1 </code><code class="r comments"># Set cutoff for inbreeding coefficient;</code></div><div class="line number9 index8 alt2"><code class="r plain">het_call &lt;- </code><code class="r functions">with</code><code class="r plain">(Sampstats, </code><code class="r functions">abs</code><code class="r plain">(hetF) &lt; het &amp; Call.rate == 1)</code></div><div class="line number10 index9 alt1"><code class="r plain">genData$SNP &lt;- genData$SNP[het_call,]</code></div><div class="line number11 index10 alt2"><code class="r plain">genData$LIP &lt;- genData$LIP[het_call,]</code></div></div></td></tr></tbody></table></div></div>
<p>Finally, we will investigate relatedness among samples using the kinship coefficient based on identity by descent (IBD).&nbsp;Please note that these functions from the package <strong>SNPRelate&nbsp;</strong>require GDS files. For this reason we first need to aggregate the .bed, .fam and .bim files from the three populations into <strong>convertGDS</strong>. The function <strong>snpgdsBED2GDS2&nbsp;</strong>creates the GDS necessary for this part of the analysis. To determine the kinship coefficient between pairs of samples we will use a subset of uncorrelated SNPs in order to have unbiased estimates. For this purpose, we will use linkage disequilibrium (LD) as a measure of correlation between markers. LD ranges from 0 to 1, the higher its value the more likely two SNPs co-segregate and therefore correlate. Here, we will utilize the subset of SNPs with LD &lt; 0.2 (<em>p</em>&nbsp;~ 12,000) to determine the IBD kinship coefficient. It took me about 2 hours to calculate the LD with the function <strong>snpgdsLDpruning</strong>, so be patient. Next, following&nbsp;<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6605/full" rel="nofollow" target="_blank">Reed et al., 2015</a>, we will exclude all samples with kinship coefficients &gt; 0.1.</p>
<div><div id="highlighter_94462" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div><div class="line number12 index11 alt1">12</div><div class="line number13 index12 alt2">13</div><div class="line number14 index13 alt1">14</div><div class="line number15 index14 alt2">15</div><div class="line number16 index15 alt1">16</div><div class="line number17 index16 alt2">17</div><div class="line number18 index17 alt1">18</div><div class="line number19 index18 alt2">19</div><div class="line number20 index19 alt1">20</div><div class="line number21 index20 alt2">21</div><div class="line number22 index21 alt1">22</div><div class="line number23 index22 alt2">23</div><div class="line number24 index23 alt1">24</div><div class="line number25 index24 alt2">25</div><div class="line number26 index25 alt1">26</div><div class="line number27 index26 alt2">27</div><div class="line number28 index27 alt1">28</div><div class="line number29 index28 alt2">29</div><div class="line number30 index29 alt1">30</div><div class="line number31 index30 alt2">31</div><div class="line number32 index31 alt1">32</div><div class="line number33 index32 alt2">33</div><div class="line number34 index33 alt1">34</div><div class="line number35 index34 alt2">35</div><div class="line number36 index35 alt1">36</div><div class="line number37 index36 alt2">37</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># LD and kinship coeff</code></div><div class="line number2 index1 alt1"><code class="r plain">ld &lt;- .2</code></div><div class="line number3 index2 alt2"><code class="r plain">kin &lt;- .1</code></div><div class="line number4 index3 alt1"><code class="r functions">snpgdsBED2GDS</code><code class="r plain">(bed.fn = </code><code class="r string">"convertGDS.bed"</code><code class="r plain">, bim.fn = </code><code class="r string">"convertGDS.bim"</code><code class="r plain">,</code></div><div class="line number5 index4 alt2"><code class="r plain">fam.fn = </code><code class="r string">"convertGDS.fam"</code><code class="r plain">, out.gdsfn = </code><code class="r string">"myGDS"</code><code class="r plain">, cvt.chr = </code><code class="r string">"char"</code><code class="r plain">)</code></div><div class="line number6 index5 alt1"><code class="r plain">genofile &lt;- </code><code class="r functions">snpgdsOpen</code><code class="r plain">(</code><code class="r string">"myGDS"</code><code class="r plain">, readonly = F)</code></div><div class="line number7 index6 alt2"><code class="r plain">gds.ids &lt;- </code><code class="r functions">read.gdsn</code><code class="r plain">(</code><code class="r functions">index.gdsn</code><code class="r plain">(genofile, </code><code class="r string">"sample.id"</code><code class="r plain">))</code></div><div class="line number8 index7 alt1"><code class="r plain">gds.ids &lt;- </code><code class="r functions">sub</code><code class="r plain">(</code><code class="r string">"-1"</code><code class="r plain">, </code><code class="r string">""</code><code class="r plain">, gds.ids)</code></div><div class="line number9 index8 alt2"><code class="r functions">add.gdsn</code><code class="r plain">(genofile, </code><code class="r string">"sample.id"</code><code class="r plain">, gds.ids, replace = T)</code></div><div class="line number10 index9 alt1"><code class="r plain">geno.sample.ids &lt;- </code><code class="r functions">rownames</code><code class="r plain">(genData$SNP)</code></div><div class="line number11 index10 alt2"><code class="r comments"># First filter for LD</code></div><div class="line number12 index11 alt1"><code class="r plain">snpSUB &lt;- </code><code class="r functions">snpgdsLDpruning</code><code class="r plain">(genofile, ld.threshold = ld,</code></div><div class="line number13 index12 alt2"><code class="r plain">sample.id = geno.sample.ids,</code></div><div class="line number14 index13 alt1"><code class="r plain">snp.id = </code><code class="r functions">colnames</code><code class="r plain">(genData$SNP))</code></div><div class="line number15 index14 alt2"><code class="r plain">snpset.ibd &lt;- </code><code class="r functions">unlist</code><code class="r plain">(snpSUB, use.names = F)</code></div><div class="line number16 index15 alt1"><code class="r comments"># And now filter for MoM</code></div><div class="line number17 index16 alt2"><code class="r plain">ibd &lt;- </code><code class="r functions">snpgdsIBDMoM</code><code class="r plain">(genofile, kinship = T,</code></div><div class="line number18 index17 alt1"><code class="r plain">sample.id = geno.sample.ids,</code></div><div class="line number19 index18 alt2"><code class="r plain">snp.id = snpset.ibd,</code></div><div class="line number20 index19 alt1"><code class="r plain">num.thread = 1)</code></div><div class="line number21 index20 alt2"><code class="r plain">ibdcoef &lt;- </code><code class="r functions">snpgdsIBDSelection</code><code class="r plain">(ibd)</code></div><div class="line number22 index21 alt1"><code class="r plain">ibdcoef &lt;- ibdcoef[ibdcoef$kinship &gt;= kin,]</code></div><div class="line number23 index22 alt2">&nbsp;</div><div class="line number24 index23 alt1"><code class="r comments"># Filter samples out</code></div><div class="line number25 index24 alt2"><code class="r plain">related.samples &lt;- </code><code class="r keyword">NULL</code></div><div class="line number26 index25 alt1"><code class="r functions">while </code><code class="r plain">(</code><code class="r functions">nrow</code><code class="r plain">(ibdcoef) &gt; 0) {</code></div><div class="line number27 index26 alt2"><code class="r comments"># count the number of occurrences of each and take the top one</code></div><div class="line number28 index27 alt1"><code class="r plain">sample.counts &lt;- </code><code class="r functions">arrange</code><code class="r plain">(</code><code class="r functions">count</code><code class="r plain">(</code><code class="r functions">c</code><code class="r plain">(ibdcoef$ID1, ibdcoef$ID2)), -freq)</code></div><div class="line number29 index28 alt2"><code class="r plain">rm.sample &lt;- sample.counts[1, </code><code class="r string">'x'</code><code class="r plain">]</code></div><div class="line number30 index29 alt1"><code class="r functions">cat</code><code class="r plain">(</code><code class="r string">"Removing sample"</code><code class="r plain">, </code><code class="r functions">as.character</code><code class="r plain">(rm.sample), </code><code class="r string">'too closely related to'</code><code class="r plain">, sample.counts[1, </code><code class="r string">'freq'</code><code class="r plain">],</code><code class="r string">'other samples.\n'</code><code class="r plain">)</code></div><div class="line number31 index30 alt2">&nbsp;</div><div class="line number32 index31 alt1"><code class="r comments"># remove from ibdcoef and add to list</code></div><div class="line number33 index32 alt2"><code class="r plain">ibdcoef &lt;- ibdcoef[ibdcoef$ID1 != rm.sample &amp; ibdcoef$ID2 != rm.sample,]</code></div><div class="line number34 index33 alt1"><code class="r plain">related.samples &lt;- </code><code class="r functions">c</code><code class="r plain">(</code><code class="r functions">as.character</code><code class="r plain">(rm.sample), related.samples)</code></div><div class="line number35 index34 alt2"><code class="r plain">}</code></div><div class="line number36 index35 alt1"><code class="r plain">genData$SNP &lt;- genData$SNP[!(</code><code class="r functions">rownames</code><code class="r plain">(genData$SNP) %</code><code class="r keyword">in</code><code class="r plain">% related.samples),]</code></div><div class="line number37 index36 alt2"><code class="r plain">genData$LIP &lt;- genData$LIP[!(</code><code class="r functions">rownames</code><code class="r plain">(genData$LIP) %</code><code class="r keyword">in</code><code class="r plain">% related.samples),]</code></div></div></td></tr></tbody></table></div></div>
<p>After pre-processing, we are left with 316 samples (110 Chinese, 100 Indian and 106 Malay) characterised by 795,668 SNP markers and 282 lipid species. Note that your sample size might differ slightly as the LD pruning procedure is stochastic.</p>
<h2>Analysis</h2>
<h3>Principal Component Analysis</h3>
<p>Now that we are done with the pre-processing, it might be a good idea to examine the largest sources of variation in the genotype data and look out for outliers or clustering patterns, using <a href="https://poissonisfish.wordpress.com/2017/01/23/principal-component-analysis-in-r/" rel="nofollow" target="_blank">Principal Component Analysis (PCA)</a>. Because we are working with S4 objects, we will be using the PCA function from <strong>SNPRelate</strong>,&nbsp;<strong>snpgdsPCA</strong>. Let’s plot the first two principal components (PCs).</p>
<div><div id="highlighter_969803" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div><div class="line number12 index11 alt1">12</div><div class="line number13 index12 alt2">13</div><div class="line number14 index13 alt1">14</div><div class="line number15 index14 alt2">15</div><div class="line number16 index15 alt1">16</div><div class="line number17 index16 alt2">17</div><div class="line number18 index17 alt1">18</div><div class="line number19 index18 alt2">19</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># PCA</code></div><div class="line number2 index1 alt1"><code class="r plain">pca &lt;- </code><code class="r functions">snpgdsPCA</code><code class="r plain">(genofile, sample.id = geno.sample.ids, snp.id = snpset.ibd, num.thread = 1)</code></div><div class="line number3 index2 alt2"><code class="r plain">pctab &lt;- </code><code class="r functions">data.frame</code><code class="r plain">(sample.id = pca$sample.id,</code></div><div class="line number4 index3 alt1"><code class="r plain">PC1 = pca$eigenvect[,1],</code></div><div class="line number5 index4 alt2"><code class="r plain">PC2 = pca$eigenvect[,2],</code></div><div class="line number6 index5 alt1"><code class="r plain">stringsAsFactors = F)</code></div><div class="line number7 index6 alt2">&nbsp;</div><div class="line number8 index7 alt1"><code class="r plain">origin &lt;- </code><code class="r functions">read.delim</code><code class="r plain">(</code><code class="r string">"countryOrigin.txt"</code><code class="r plain">, sep = </code><code class="r string">"\t"</code><code class="r plain">)</code></div><div class="line number9 index8 alt2"><code class="r plain">origin &lt;- origin[</code><code class="r functions">match</code><code class="r plain">(pca$sample.id, origin$sample.id),]</code></div><div class="line number10 index9 alt1">&nbsp;</div><div class="line number11 index10 alt2"><code class="r plain">pcaCol &lt;- </code><code class="r functions">rep</code><code class="r plain">(</code><code class="r functions">rgb</code><code class="r plain">(0,0,0,.3), </code><code class="r functions">length</code><code class="r plain">(pca$sample.id)) </code><code class="r comments"># Set black for chinese</code></div><div class="line number12 index11 alt1"><code class="r plain">pcaCol[origin$Country == </code><code class="r string">"I"</code><code class="r plain">] &lt;- </code><code class="r functions">rgb</code><code class="r plain">(1,0,0,.3) </code><code class="r comments"># red for indian</code></div><div class="line number13 index12 alt2"><code class="r plain">pcaCol[origin$Country == </code><code class="r string">"M"</code><code class="r plain">] &lt;- </code><code class="r functions">rgb</code><code class="r plain">(0,.7,0,.3) </code><code class="r comments"># green for malay</code></div><div class="line number14 index13 alt1">&nbsp;</div><div class="line number15 index14 alt2"><code class="r functions">png</code><code class="r plain">(</code><code class="r string">"PCApopulation.png"</code><code class="r plain">, width = 500, height = 500)</code></div><div class="line number16 index15 alt1"><code class="r functions">plot</code><code class="r plain">(pctab$PC1, pctab$PC2, xlab = </code><code class="r string">"PC1"</code><code class="r plain">, ylab = </code><code class="r string">"PC2"</code><code class="r plain">, col = pcaCol, pch = 16)</code></div><div class="line number17 index16 alt2"><code class="r functions">abline</code><code class="r plain">(h = 0, v = 0, lty = 2, col = </code><code class="r string">"grey"</code><code class="r plain">)</code></div><div class="line number18 index17 alt1"><code class="r functions">legend</code><code class="r plain">(</code><code class="r string">"top"</code><code class="r plain">, legend = </code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">"Chinese"</code><code class="r plain">, </code><code class="r string">"Indian"</code><code class="r plain">, </code><code class="r string">"Malay"</code><code class="r plain">), col = 1:3, pch = 16, bty = </code><code class="r string">"n"</code><code class="r plain">)</code></div><div class="line number19 index18 alt2"><code class="r functions">dev.off</code><code class="r plain">()</code></div></div></td></tr></tbody></table></div></div>
<h2><img src="./Genome-wide association studies in R _ R-bloggers_files/pcapopulation1.png" data-attachment-id="3242" data-permalink="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/pcapopulation/" data-orig-file="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456" data-orig-size="500,500" data-comments-opened="1" data-image-title="PCApopulation" data-image-description="" data-medium-file="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456?w=300" data-large-file="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456?w=500" class="alignnone size-full wp-image-3242" alt="PCApopulation" sizes="(max-width: 500px) 100vw, 500px" data-lazy-loaded="true" style="display: inline;" width="456" height="456" originalw="456" src-orig="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456" scale="1.375"><noscript>&lt;img data-attachment-id="3242" data-permalink="<a class="vglnk" href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/pcapopulation/" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>09</span><span>/</span><span>genome</span><span>-</span><span>wide</span><span>-</span><span>association</span><span>-</span><span>studies</span><span>-</span><span>in</span><span>-</span><span>r</span><span>/</span><span>pcapopulation</span><span>/</span></a>" data-orig-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" data-orig-size="500,500" data-comments-opened="1" data-image-meta="{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}" data-image-title="PCApopulation" data-image-description="" data-medium-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>300</span></a>" data-large-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456?w=500" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>500</span></a>" class="alignnone size-full wp-image-3242" src="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" alt="PCApopulation" srcset_temp="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span></a> 500w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=150" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>150</span></a> 150w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/pcapopulation1.png?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>pcapopulation1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>300</span></a> 300w" sizes="(max-width: 500px) 100vw, 500px"   data-recalc-dims="1" /&gt;</noscript></h2>
<p>As expected, the&nbsp;795,668 SNP markers clearly delineate the three populations. The results also suggest that Chinese and Malay are closer to each other than to Indian (this observation would be much better addressed with e.g. hierarchical clustering). Also, no obvious outliers are identified with the first two PCs. All good to proceed to GWA.</p>
<h3>Genome-Wide Association</h3>
<p>Finally, the&nbsp;<span style="line-height:inherit;">pièce de résistance</span>. From the set of 282 lipids species I chose cholesterol, one of the most familiar, as the trait of interest. You are completely free to select a different lipid species and proceed. We are going to use the GWA function provided in&nbsp;<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6605/full" rel="nofollow" target="_blank">Reed et al., 2015</a> with some minor modifications. I recommend you to open the script <strong>GWASfunction.R</strong> &nbsp;and skim through. This is an excellent, well documented parallelized implementation. Note that the&nbsp;<strong>glm </strong>function&nbsp;is used to determine the significance of association between each SNP and the trait of interest. In case you are wondering, <strong>glm</strong> is much more versatile than <strong>lm</strong> since it conducts Gaussian, Poisson, binomial and multinomial regression / classification, depending on how your trait of interest is distributed (all lipids in the phenotype file are Gaussian). This GWA function will not create a variable, but rather write a .txt summary table listing the coefficient estimate <img src="./Genome-wide association studies in R _ R-bloggers_files/latex.php" alt="\beta " title="\beta " class="latex" data-lazy-loaded="true" style="display: inline;" width="10" height="14" src-orig="https://s0.wp.com/latex.php?latex=%5Cbeta+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=%5Cbeta+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=%</span><span>5Cbeta</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="\beta " title="\beta " class="latex" /&gt;</noscript>, <em>t</em> and the corresponding&nbsp;<em>P</em>-value for each SNP, alongside with the corresponding genomic coordinates.&nbsp;Running the GWA function took me approximately 1.5 hours with my&nbsp;MacBook Pro mid-2012 (8Gb,&nbsp;2.9 GHz Intel Core i7).</p>
<div><div id="highlighter_597369" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div><div class="line number11 index10 alt2">11</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># Choose trait for association analysis, use colnames(genData$LIP) for listing</code></div><div class="line number2 index1 alt1"><code class="r comments"># NOTE: Ignore the first column of genData$LIP (gender)</code></div><div class="line number3 index2 alt2"><code class="r plain">target &lt;- </code><code class="r string">"Cholesterol"</code></div><div class="line number4 index3 alt1">&nbsp;</div><div class="line number5 index4 alt2"><code class="r plain">phenodata &lt;- </code><code class="r functions">data.frame</code><code class="r plain">(</code><code class="r string">"id"</code> <code class="r plain">= </code><code class="r functions">rownames</code><code class="r plain">(genData$LIP),</code></div><div class="line number6 index5 alt1"><code class="r string">"phenotype"</code> <code class="r plain">= </code><code class="r functions">scale</code><code class="r plain">(genData$LIP[,target]), stringsAsFactors = F)</code></div><div class="line number7 index6 alt2">&nbsp;</div><div class="line number8 index7 alt1"><code class="r comments"># Conduct GWAS (will take a while)</code></div><div class="line number9 index8 alt2"><code class="r plain">start &lt;- </code><code class="r functions">Sys.time</code><code class="r plain">()</code></div><div class="line number10 index9 alt1"><code class="r functions">GWAA</code><code class="r plain">(genodata = genData$SNP, phenodata = phenodata, filename = </code><code class="r functions">paste</code><code class="r plain">(target, </code><code class="r string">".txt"</code><code class="r plain">, sep = </code><code class="r string">""</code><code class="r plain">))</code></div><div class="line number11 index10 alt2"><code class="r functions">Sys.time</code><code class="r plain">() - start </code><code class="r comments"># benchmark</code></div></div></td></tr></tbody></table></div></div>
<p>Once finished, we can visualize the results using the so-called Manhattan plots. All we need is to load the .txt summary table written in the previous step, add a column with <img src="./Genome-wide association studies in R _ R-bloggers_files/latex(7).php" alt="-\log_{10} (P) " title="-\log_{10} (P) " class="latex" data-lazy-loaded="true" style="display: inline;" width="72" height="19" src-orig="https://s0.wp.com/latex.php?latex=-%5Clog_%7B10%7D+%28P%29+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" scale="1.375"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=-%5Clog_%7B10%7D+%28P%29+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=-%</span><span>5Clog</span><span>_%</span><span>7B10</span><span>%</span><span>7D</span><span>+%</span><span>28P</span><span>%</span><span>29</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="-\log_{10} (P) " title="-\log_{10} (P) " class="latex" /&gt;</noscript> and plot these significance estimates against the genomic coordinates of all SNPs.</p>
<div><div id="highlighter_227894" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div><div class="line number5 index4 alt2">5</div><div class="line number6 index5 alt1">6</div><div class="line number7 index6 alt2">7</div><div class="line number8 index7 alt1">8</div><div class="line number9 index8 alt2">9</div><div class="line number10 index9 alt1">10</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># Manhattan plot</code></div><div class="line number2 index1 alt1"><code class="r plain">GWASout &lt;- </code><code class="r functions">read.table</code><code class="r plain">(</code><code class="r functions">paste</code><code class="r plain">(target, </code><code class="r string">".txt"</code><code class="r plain">, sep = </code><code class="r string">""</code><code class="r plain">), header = T, colClasses = </code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">"character"</code><code class="r plain">, </code><code class="r functions">rep</code><code class="r plain">(</code><code class="r string">"numeric"</code><code class="r plain">,4)))</code></div><div class="line number3 index2 alt2"><code class="r plain">GWASout$type &lt;- </code><code class="r functions">rep</code><code class="r plain">(</code><code class="r string">"typed"</code><code class="r plain">, </code><code class="r functions">nrow</code><code class="r plain">(GWASout))</code></div><div class="line number4 index3 alt1"><code class="r plain">GWASout$Neg_logP &lt;- -</code><code class="r functions">log10</code><code class="r plain">(GWASout$p.value)</code></div><div class="line number5 index4 alt2"><code class="r plain">GWASout &lt;- </code><code class="r functions">merge</code><code class="r plain">(GWASout, genData$MAP[,</code><code class="r functions">c</code><code class="r plain">(</code><code class="r string">"SNP"</code><code class="r plain">, </code><code class="r string">"chr"</code><code class="r plain">, </code><code class="r string">"position"</code><code class="r plain">)])</code></div><div class="line number6 index5 alt1"><code class="r plain">GWASout &lt;- GWASout[</code><code class="r functions">order</code><code class="r plain">(GWASout$Neg_logP, decreasing = T),]</code></div><div class="line number7 index6 alt2">&nbsp;</div><div class="line number8 index7 alt1"><code class="r functions">png</code><code class="r plain">(</code><code class="r functions">paste</code><code class="r plain">(target, </code><code class="r string">".png"</code><code class="r plain">, sep = </code><code class="r string">""</code><code class="r plain">), height = 500,width = 1000)</code></div><div class="line number9 index8 alt2"><code class="r functions">GWAS_Manhattan</code><code class="r plain">(GWASout)</code></div><div class="line number10 index9 alt1"><code class="r functions">dev.off</code><code class="r plain">()</code></div></div></td></tr></tbody></table></div></div>
<p>In addition, a full (resp. dashed) line indicate the levels of Bonferroni-adjusted <img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans.gif" data-lazy-src="https://s0.wp.com/latex.php?latex=%5Calpha+%3D+0.05+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" alt="\alpha = 0.05 " title="\alpha = 0.05 " class="latex"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=%5Calpha+%3D+0.05+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=%</span><span>5Calpha</span><span>+%</span><span>3D</span><span>+</span><span>0</span><span>.</span><span>05</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="\alpha = 0.05 " title="\alpha = 0.05 " class="latex" /&gt;</noscript> for 1,000,000 (resp. 10,000) tests.<img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans(1).gif" data-lazy-src="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456" data-attachment-id="3243" data-permalink="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/cholesterol/" data-orig-file="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456" data-orig-size="1000,500" data-comments-opened="1" data-image-title="Cholesterol" data-image-description="" data-medium-file="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456?w=300" data-large-file="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456?w=1000" class="alignnone size-full wp-image-3243" alt="Cholesterol" sizes="(max-width: 1000px) 100vw, 1000px" data-recalc-dims="1"><noscript>&lt;img data-attachment-id="3243" data-permalink="<a class="vglnk" href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/cholesterol/" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>09</span><span>/</span><span>genome</span><span>-</span><span>wide</span><span>-</span><span>association</span><span>-</span><span>studies</span><span>-</span><span>in</span><span>-</span><span>r</span><span>/</span><span>cholesterol</span><span>/</span></a>" data-orig-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" data-orig-size="1000,500" data-comments-opened="1" data-image-meta="{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}" data-image-title="Cholesterol" data-image-description="" data-medium-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>300</span></a>" data-large-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456?w=1000" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>1000</span></a>" class="alignnone size-full wp-image-3243" src="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" alt="Cholesterol" srcset_temp="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span></a> 1000w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=150" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>150</span></a> 150w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>300</span></a> 300w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol1.png?w=768" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol1</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>768</span></a> 768w" sizes="(max-width: 1000px) 100vw, 1000px"   data-recalc-dims="1" /&gt;</noscript></p>
<p>We see that a total of four SNPs pass the ‘relaxed’ Bonferroni threshold (none passes the ‘hard’ threshold). These are SNPs rs7527051 (Chr. 1), rs12140539 (co-localized with the first, Chr. 1), rs9509213 (Chr. 13) and rs2250402 (Chr. 15).</p>
<p>Before proceeding with these four hits, it is helpful to constrast the distribution of the resulting <em>P</em>-values against that expected by chance, as to ensure there is no confounding systemic bias. This is easily addressed with a quantile-quantile (Q-Q) plot. As the name suggests, it compares two distributions based on their quantiles. In the present case, we want to contrast the distribution of our <em>t</em> statistics against that obtained by chance. If two distributions are identical in shape, the Q-Q plot will display a <img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans.gif" data-lazy-src="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" alt="x = y " title="x = y " class="latex"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>x</span><span>+%</span><span>3D</span><span>+</span><span>y</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="x = y " title="x = y " class="latex" /&gt;</noscript> line. Therefore, the Q-Q plot from a reliable GWA analysis will display a <img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans.gif" data-lazy-src="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" alt="x = y " title="x = y " class="latex"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>x</span><span>+%</span><span>3D</span><span>+</span><span>y</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="x = y " title="x = y " class="latex" /&gt;</noscript> line with only few deviating values that are suggestive of association. Otherwise, if the line is shifted up or down&nbsp;the GWA analysis might be impaired by confounding factors.<strong>&nbsp;</strong>We will draw a Q-Q plot using the function <strong>estlambda</strong> from the package <strong>GenABEL</strong>.</p>
<div><div id="highlighter_184610" class="syntaxhighlighter  r"><table border="0" cellpadding="0" cellspacing="0"><tbody><tr><td class="gutter"><div class="line number1 index0 alt2">1</div><div class="line number2 index1 alt1">2</div><div class="line number3 index2 alt2">3</div><div class="line number4 index3 alt1">4</div></td><td class="code"><div class="container"><div class="line number1 index0 alt2"><code class="r comments"># QQ plot using GenABEL estlambda function</code></div><div class="line number2 index1 alt1"><code class="r functions">png</code><code class="r plain">(</code><code class="r functions">paste</code><code class="r plain">(target, </code><code class="r string">"_QQplot.png"</code><code class="r plain">, sep = </code><code class="r string">""</code><code class="r plain">), width = 500, height = 500)</code></div><div class="line number3 index2 alt2"><code class="r plain">lambda &lt;- </code><code class="r functions">estlambda</code><code class="r plain">(GWASout$t.value**2, plot = T, method = </code><code class="r string">"median"</code><code class="r plain">)</code></div><div class="line number4 index3 alt1"><code class="r functions">dev.off</code><code class="r plain">()</code></div></div></td></tr></tbody></table></div></div>
<p><img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans(1).gif" data-lazy-src="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456" data-attachment-id="3244" data-permalink="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/cholesterol_qqplot/" data-orig-file="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456" data-orig-size="500,500" data-comments-opened="1" data-image-title="Cholesterol_QQplot" data-image-description="" data-medium-file="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456?w=300" data-large-file="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456?w=500" class="alignnone size-full wp-image-3244" alt="Cholesterol_QQplot" sizes="(max-width: 500px) 100vw, 500px" data-recalc-dims="1"><noscript>&lt;img data-attachment-id="3244" data-permalink="<a class="vglnk" href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/cholesterol_qqplot/" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>09</span><span>/</span><span>genome</span><span>-</span><span>wide</span><span>-</span><span>association</span><span>-</span><span>studies</span><span>-</span><span>in</span><span>-</span><span>r</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot</span><span>/</span></a>" data-orig-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" data-orig-size="500,500" data-comments-opened="1" data-image-meta="{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}" data-image-title="Cholesterol_QQplot" data-image-description="" data-medium-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>300</span></a>" data-large-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456?w=500" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>500</span></a>" class="alignnone size-full wp-image-3244" src="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" alt="Cholesterol_QQplot" srcset_temp="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span></a> 500w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=150" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>150</span></a> 150w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/cholesterol_qqplot2.png?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>cholesterol</span><span>_</span><span>qqplot2</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>300</span></a> 300w" sizes="(max-width: 500px) 100vw, 500px"   data-recalc-dims="1" /&gt;</noscript></p>
<p>The resulting Q-Q plot clearly depicts a trend line (<img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans.gif" data-lazy-src="https://s0.wp.com/latex.php?latex=%5Clambda+%3D+0.99+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" alt="\lambda = 0.99 " title="\lambda = 0.99 " class="latex"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=%5Clambda+%3D+0.99+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=%</span><span>5Clambda</span><span>+%</span><span>3D</span><span>+</span><span>0</span><span>.</span><span>99</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="\lambda = 0.99 " title="\lambda = 0.99 " class="latex" /&gt;</noscript>, red) overlapping with <img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans.gif" data-lazy-src="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" alt="x = y " title="x = y " class="latex"><noscript>&lt;img src="<a class="vglnk" href="https://s0.wp.com/latex.php?latex=x+%3D+y+&amp;bg=f9f9f9&amp;%23038;fg=555555&amp;%23038;s=1" rel="nofollow"><span>https</span><span>://</span><span>s0</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>latex</span><span>.</span><span>php</span><span>?</span><span>latex</span><span>=</span><span>x</span><span>+%</span><span>3D</span><span>+</span><span>y</span><span>+&amp;</span><span>bg</span><span>=</span><span>f9f9f9</span><span>&amp;%</span><span>23038</span><span>;</span><span>fg</span><span>=</span><span>555555</span><span>&amp;%</span><span>23038</span><span>;</span><span>s</span><span>=</span><span>1</span></a>" alt="x = y " title="x = y " class="latex" /&gt;</noscript> (black) and a slight deviation in the right tail, so we can be confident about our results.</p>
<h3>Functional insights into candidate markers</h3>
<p>Finally, we will try to find the functional relevance of these four candidate SNPs by searching for genes in their vicinity, using the&nbsp;<a href="https://genome.ucsc.edu/" rel="nofollow" target="_blank">USCS&nbsp;Genome Browser</a>&nbsp;(enter Genome Browser, insert the SNP ID in the text box, enter and zoom out).&nbsp;I found that&nbsp;rs9509213 (Chr. 13) lands right on <em>CRYL1</em> (crystalline lambda 1, intron sequence), rendering it an interesting candidate for follow-up studies.</p>
<p><img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans(1).gif" data-lazy-src="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456" data-attachment-id="3817" data-permalink="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/screen-shot-2017-10-08-at-17-58-52/" data-orig-file="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456" data-orig-size="1241,567" data-comments-opened="1" data-image-title="Screen Shot 2017-10-08 at 17.58.52" data-image-description="" data-medium-file="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456?w=300" data-large-file="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456?w=1024" class="alignnone size-full wp-image-3817" alt="Screen Shot 2017-10-08 at 17.58.52" sizes="(max-width: 1108px) 100vw, 1108px" data-recalc-dims="1"><noscript>&lt;img data-attachment-id="3817" data-permalink="<a class="vglnk" href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/screen-shot-2017-10-08-at-17-58-52/" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>09</span><span>/</span><span>genome</span><span>-</span><span>wide</span><span>-</span><span>association</span><span>-</span><span>studies</span><span>-</span><span>in</span><span>-</span><span>r</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>/</span></a>" data-orig-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" data-orig-size="1241,567" data-comments-opened="1" data-image-meta="{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}" data-image-title="Screen Shot 2017-10-08 at 17.58.52" data-image-description="" data-medium-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>300</span></a>" data-large-file="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456?w=1024" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span><span>?</span><span>w</span><span>=</span><span>1024</span></a>" class="alignnone size-full wp-image-3817" src="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a>" alt="Screen Shot 2017-10-08 at 17.58.52" srcset_temp="<a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=456" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>456</span></a> 1108w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=150" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>150</span></a> 150w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=300" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>300</span></a> 300w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=768" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>768</span></a> 768w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png?w=1024" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span><span>?</span><span>w</span><span>=</span><span>1024</span></a> 1024w, <a class="vglnk" href="https://poissonisfish.files.wordpress.com/2017/10/screen-shot-2017-10-08-at-17-58-52.png" rel="nofollow"><span>https</span><span>://</span><span>poissonisfish</span><span>.</span><span>files</span><span>.</span><span>wordpress</span><span>.</span><span>com</span><span>/</span><span>2017</span><span>/</span><span>10</span><span>/</span><span>screen</span><span>-</span><span>shot</span><span>-</span><span>2017</span><span>-</span><span>10</span><span>-</span><span>08</span><span>-</span><span>at</span><span>-</span><span>17</span><span>-</span><span>58</span><span>-</span><span>52</span><span>.</span><span>png</span></a> 1241w" sizes="(max-width: 1108px) 100vw, 1108px"   data-recalc-dims="1" /&gt;</noscript></p>
<p>The SNP&nbsp;rs9509213 is shown as a black text box in the bottom of the figure, and its coordinates are highlighted by a vertical yellow line. The <em>CRYL1</em> gene model is shown on top, below the chromosome model (topmost).</p>
<p>Interestingly, there is a recent <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5400530/" rel="nofollow" target="_blank">publication</a> that found ‘<em>POE, HP, and CRYL1 have all been associated with Alzheimer’s Disease, the pathology of which involves lipid and cholesterol pathways.’</em>.</p>
<p>Finally, I would like to remark on the need of experimental validation of gene candidates to validate the resulting SNP-trait associations. In this particular case, one could test the effect of overexpressing or knocking-out the ortholog of <em>CRYL1</em> in cholesterol levels of mice.</p>
<h3>Wrap-up</h3>
<p>To sum up, we have covered some of the most fundamental aspects of GWA analysis:</p>
<ul>
<li>Pre-processing of genotype data based on call rates, MAF, kinship and&nbsp;heterozygosity</li>
<li>Investigation of population structure with PCA of the genotype data</li>
<li>GWA and visualisation with Manhattan plots</li>
<li>Q-Q plots</li>
<li>Functional insights into the resulting candidate SNPs</li>
</ul>
<p>I did not cover many other aspects of GWA, such as fine-mapping using LD plots. I encourage you to choose a different lipid species from the phenotype data, run the analysis and interpret the results. Also noteworthy, the state-of-the-art of GWA is now almost-fully based on <a href="http://www.nature.com/ng/journal/v42/n4/full/ng.546.html" rel="nofollow" target="_blank">mixed-effect linear models</a>&nbsp;(MLM) that consider kinship and cryptic relatedness as random factors, hence much more powerful that <strong>glm</strong>. My next post might cover MLM.</p>
<p>I would like to thank&nbsp;<a href="https://www.nature.com/articles/s41467-017-00413-x" rel="nofollow" target="_blank">Saw et al., 2017</a>&nbsp;for providing unrestricted access to their <a href="http://phg.nus.edu.sg/#iomics" rel="nofollow" target="_blank">omics data</a>. If it wasn’t for such conscientious scientists and their transparent research endeavours this whole tutorial would simply not exist. I also reiterate that this tutorial is largely based on the guide outlined in&nbsp;<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6605/full" rel="nofollow" target="_blank">Reed et al., 2015</a>, an excellent reference along with <a href="https://www.ncbi.nlm.nih.gov/pubmed/23711206" rel="nofollow" target="_blank">Sikorska et al., 2013</a>.</p>
<p>Have fun and keep your cholesterol in check. Any feedback or suggestions are welcome!</p>
<p>  <a rel="nofollow" href="http://feeds.wordpress.com/1.0/gocomments/poissonisfish.wordpress.com/3115/" target="_blank"><img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans.gif" data-lazy-src="http://feeds.wordpress.com/1.0/comments/poissonisfish.wordpress.com/3115/" alt="" border="0"><noscript>&lt;img alt="" border="0" src="http://feeds.wordpress.com/1.0/comments/poissonisfish.wordpress.com/3115/" /&gt;</noscript></a> <img src="./Genome-wide association studies in R _ R-bloggers_files/1x1.trans(1).gif" data-lazy-src="https://i1.wp.com/pixel.wp.com/b.gif?resize=1%2C1&amp;ssl=1" alt="" border="0" data-recalc-dims="1"><noscript>&lt;img alt="" border="0" src="<a class="vglnk" href="https://i1.wp.com/pixel.wp.com/b.gif?resize=1%2C1&amp;#038;ssl=1" rel="nofollow"><span>https</span><span>://</span><span>i1</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>pixel</span><span>.</span><span>wp</span><span>.</span><span>com</span><span>/</span><span>b</span><span>.</span><span>gif</span><span>?</span><span>resize</span><span>=</span><span>1</span><span>%</span><span>2C1</span><span>&amp;#</span><span>038</span><span>;</span><span>ssl</span><span>=</span><span>1</span></a>" data-recalc-dims="1" /&gt;</noscript></p>

		<script type="text/javascript">
		  var vglnk = { key: '949efb41171ac6ec1bf7f206d57e90b8' };

		  (function(d, t) {
			var s = d.createElement(t); s.type = 'text/javascript'; s.async = true;
			s.src = '//cdn.viglink.com/api/vglnk.js';
			var r = d.getElementsByTagName(t)[0]; r.parentNode.insertBefore(s, r);
		  }(document, 'script'));
		</script>		
		
<div id="jp-relatedposts" class="jp-relatedposts" style="display: block;">
	<h3 class="jp-relatedposts-headline"><em>Related</em></h3>
<div class="jp-relatedposts-items jp-relatedposts-items-visual jp-relatedposts-grid "><div class="jp-relatedposts-post jp-relatedposts-post0 jp-relatedposts-post-nothumbs" data-post-id="16546" data-post-format="false"><a class="jp-relatedposts-post-a jp-relatedposts-post-aoverlay" href="https://www.r-bloggers.com/mixed-linear-model-approach-adapted-for-genome-wide-association-studies/" title="Mixed linear model approach adapted for genome-wide association studies

A few weeks ago I covered an R package for efficient mixed model regression that is capable of simultaneously accounting for both population stratification and relatedness to compute unbiased estimates of standard errors and p-values for genetic associ..." rel="nofollow" data-origin="155868" data-position="0"></a><h4 class="jp-relatedposts-post-title"><a class="jp-relatedposts-post-a" href="https://www.r-bloggers.com/mixed-linear-model-approach-adapted-for-genome-wide-association-studies/" title="Mixed linear model approach adapted for genome-wide association studies

A few weeks ago I covered an R package for efficient mixed model regression that is capable of simultaneously accounting for both population stratification and relatedness to compute unbiased estimates of standard errors and p-values for genetic associ..." rel="nofollow" data-origin="155868" data-position="0">Mixed linear model approach adapted for genome-wide association studies</a></h4><p class="jp-relatedposts-post-excerpt" style="max-height: 7.14286em;">A few weeks ago I covered an R package for efficient mixed model regression that is capable of simultaneously accounting for both population stratification and relatedness to compute unbiased estimates of standard errors and p-values for genetic associ...</p><p class="jp-relatedposts-post-date" style="display: block;">May 6, 2010</p><p class="jp-relatedposts-post-context">In "R bloggers"</p></div><div class="jp-relatedposts-post jp-relatedposts-post1 jp-relatedposts-post-nothumbs" data-post-id="62156" data-post-format="false"><a class="jp-relatedposts-post-a jp-relatedposts-post-aoverlay" href="https://www.r-bloggers.com/an-easy-way-to-manage-your-genome-wide-association-data-genabel-package/" title="An easy way to manage your genome-wide-association data: GenABEL package.

Here is a little overview on GenABEL library developed by Yurii Aulchenko (www.genabel.org/). GenABEL is a full-featured R library for dealing with Genome-Wide Association analysis of binary and quantitative traits. Compared to the ‘genetics’ package and many other tools, GenABEL … Continue reading →" rel="nofollow" data-origin="155868" data-position="1"></a><h4 class="jp-relatedposts-post-title"><a class="jp-relatedposts-post-a" href="https://www.r-bloggers.com/an-easy-way-to-manage-your-genome-wide-association-data-genabel-package/" title="An easy way to manage your genome-wide-association data: GenABEL package.

Here is a little overview on GenABEL library developed by Yurii Aulchenko (www.genabel.org/). GenABEL is a full-featured R library for dealing with Genome-Wide Association analysis of binary and quantitative traits. Compared to the ‘genetics’ package and many other tools, GenABEL … Continue reading →" rel="nofollow" data-origin="155868" data-position="1">An easy way to manage your genome-wide-association data: GenABEL package.</a></h4><p class="jp-relatedposts-post-excerpt" style="max-height: 7.14286em;">Here is a little overview on GenABEL library developed by Yurii Aulchenko (<a class="vglnk" href="http://www.genabel.org/" rel="nofollow"><span>www</span><span>.</span><span>genabel</span><span>.</span><span>org</span><span>/</span></a>). GenABEL is a full-featured R library for dealing with Genome-Wide Association analysis of binary and quantitative traits. Compared to the ‘genetics’ package and many other tools, GenABEL … Continue reading →</p><p class="jp-relatedposts-post-date" style="display: block;">June 12, 2012</p><p class="jp-relatedposts-post-context">In "R bloggers"</p></div><div class="jp-relatedposts-post jp-relatedposts-post2 jp-relatedposts-post-nothumbs" data-post-id="144522" data-post-format="false"><a class="jp-relatedposts-post-a jp-relatedposts-post-aoverlay" href="https://www.r-bloggers.com/how-to-build-a-shiny-app-for-disease-trait-associated-locations-of-the-human-genome/" title="How to build a Shiny app for disease- &amp;amp; trait-associated locations of the human genome

This app is based on the gwascat R package and its ebicat38 database and shows trait-associated SNP locations of the human genome. You can visualize and compare the genomic locations of up to 8 traits simultaneously. The National Human Genome Research..." rel="nofollow" data-origin="155868" data-position="2"></a><h4 class="jp-relatedposts-post-title"><a class="jp-relatedposts-post-a" href="https://www.r-bloggers.com/how-to-build-a-shiny-app-for-disease-trait-associated-locations-of-the-human-genome/" title="How to build a Shiny app for disease- &amp;amp; trait-associated locations of the human genome

This app is based on the gwascat R package and its ebicat38 database and shows trait-associated SNP locations of the human genome. You can visualize and compare the genomic locations of up to 8 traits simultaneously. The National Human Genome Research..." rel="nofollow" data-origin="155868" data-position="2">How to build a Shiny app for disease- &amp; trait-associated locations of the human genome</a></h4><p class="jp-relatedposts-post-excerpt" style="max-height: 7.14286em;">This app is based on the gwascat R package and its ebicat38 database and shows trait-associated SNP locations of the human genome. You can visualize and compare the genomic locations of up to 8 traits simultaneously. The National Human Genome Research...</p><p class="jp-relatedposts-post-date" style="display: block;">December 17, 2016</p><p class="jp-relatedposts-post-context">In "R bloggers"</p></div></div></div><aside class="mashsb-container mashsb-main mashsb-stretched"><div class="mashsb-box"><div class="mashsb-count mash-small" style="float:left;"><div class="counts mashsbcount">20</div><span class="mashsb-sharetext">SHARES</span></div><div class="mashsb-buttons"><a class="mashicon-facebook mash-small mash-center mashsb-noshadow" href="https://www.facebook.com/sharer.php?u=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F" target="_blank" rel="nofollow"><span class="icon"></span><span class="text">Share</span></a><a class="mashicon-twitter mash-small mash-center mashsb-noshadow" href="https://twitter.com/intent/tweet?text=Genome-wide%20association%20studies%20in%20R&amp;url=https://www.r-bloggers.com/genome-wide-association-studies-in-r/&amp;via=Rbloggers" target="_blank" rel="nofollow"><span class="icon"></span><span class="text">Tweet</span></a><div class="onoffswitch2 mash-small mashsb-noshadow" style="display:none;"></div></div>
            </div>
                <div style="clear:both;"></div></aside>
            <!-- Share buttons by mashshare.net - Version: 3.4.4-->
<p class="syndicated-attribution">
</p><div style="border: 1px solid; background: none repeat scroll 0 0 #EDEDED; margin: 1px; font-size: 13px;">
<div style="text-align: center;">To <strong>leave a comment</strong> for the author, please follow the link and comment on their blog: <strong><a href="https://poissonisfish.wordpress.com/2017/10/09/genome-wide-association-studies-in-r/"> poissonisfish</a></strong>.</div>
<hr>
<a href="https://www.r-bloggers.com/" rel="nofollow">R-bloggers.com</a> offers <strong><a href="https://feedburner.google.com/fb/a/mailverify?uri=RBloggers" rel="nofollow">daily e-mail updates</a></strong> about <a title="The R Project for Statistical Computing" href="https://www.r-project.org/" rel="nofollow">R</a> news and <a title="R tutorials" href="https://www.r-bloggers.com/search/tutorial" rel="nofollow">tutorials</a> on topics such as: <a title="Data science" href="https://www.r-bloggers.com/search/data%20science" rel="nofollow">Data science</a>, <a title="Big Data" href="https://www.r-bloggers.com/search/Big%20Data" rel="nofollow">Big Data, </a><a title="R jobs" href="https://www.r-users.com/" rel="nofollow">R jobs</a>, visualization (<a title="ggplot and ggplot2 tutorials" href="https://www.r-bloggers.com/search/ggplot2" rel="nofollow">ggplot2</a>, <a title="Boxplots using lattice and ggplot2 tutorials" href="https://www.r-bloggers.com/search/boxplot" rel="nofollow">Boxplots</a>, <a title="Maps and gis" href="https://www.r-bloggers.com/search/map" rel="nofollow">maps</a>, <a title="Animation in R" href="https://www.r-bloggers.com/search/animation" rel="nofollow">animation</a>), programming (<a title="RStudio IDE for R" href="https://www.r-bloggers.com/search/RStudio" rel="nofollow">RStudio</a>, <a title="Sweave and literate programming" href="https://www.r-bloggers.com/search/sweave" rel="nofollow">Sweave</a>, <a title="LaTeX in R" href="https://www.r-bloggers.com/search/LaTeX" rel="nofollow">LaTeX</a>, <a title="SQL and databases" href="https://www.r-bloggers.com/search/SQL" rel="nofollow">SQL</a>, <a title="Eclipse IDE for R" href="https://www.r-bloggers.com/search/eclipse" rel="nofollow">Eclipse</a>, <a title="git and github, Version Control System" href="https://www.r-bloggers.com/search/git" rel="nofollow">git</a>, <a title="Large data in R using Hadoop" href="https://www.r-bloggers.com/search/hadoop" rel="nofollow">hadoop</a>, <a title="Web Scraping of google, facebook, yahoo, twitter and more using R" href="https://www.r-bloggers.com/search/Web+Scraping" rel="nofollow">Web Scraping</a>) statistics (<a title="Regressions and ANOVA analysis tutorials" href="https://www.r-bloggers.com/search/regression" rel="nofollow">regression</a>, <a title="principal component analysis tutorial" href="https://www.r-bloggers.com/search/PCA" rel="nofollow">PCA</a>, <a title="Time series" href="https://www.r-bloggers.com/search/time+series" rel="nofollow">time series</a>, <a title="finance trading" href="https://www.r-bloggers.com/search/trading" rel="nofollow">trading</a>) and more...
</div><p></p><hr><hr>
<div style="border: 1px solid #EB9349; background: none repeat scroll 0 0 #FDEADA; text-align: center; margin: 10px; font-size: 16px;">
If you got this far, why not <strong><u>subscribe for updates</u> </strong>from the site?  Choose your flavor: <a href="http://feedburner.google.com/fb/a/mailverify?uri=RBloggers" rel="nofollow">e-mail</a>, <a href="https://twitter.com/#!/rbloggers" rel="nofollow">twitter</a>, <a href="http://feeds.feedburner.com/RBloggers" rel="nofollow">RSS</a>, or <a href="http://www.facebook.com/pages/R-bloggers/191414254890" rel="nofollow">facebook</a>...
</div><div class="social4i" style="height:29px;"><div class="social4in" style="height:29px;float: left;"><div class="socialicons s4fblike" style="float:left;margin-right: 10px;"><div class="fb-like fb_iframe_widget" data-href="https://www.r-bloggers.com/genome-wide-association-studies-in-r/" data-send="true" data-layout="button_count" data-width="100" data-height="21" data-show-faces="false" fb-xfbml-state="rendered" fb-iframe-plugin-query="app_id=&amp;container_width=0&amp;height=21&amp;href=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F&amp;layout=button_count&amp;locale=en_US&amp;sdk=joey&amp;send=true&amp;show_faces=false&amp;width=100"><span style="vertical-align: bottom; width: 118px; height: 20px;"><iframe name="f398886bae2f908" width="100px" height="21px" frameborder="0" allowtransparency="true" allowfullscreen="true" scrolling="no" title="fb:like Facebook Social Plugin" src="./Genome-wide association studies in R _ R-bloggers_files/like(1).html" style="border: none; visibility: visible; width: 118px; height: 20px;" class=""></iframe></span></div></div><div class="socialicons s4linkedin" style="float:left;margin-right: 10px;"><span class="IN-widget" style="line-height: 1; vertical-align: baseline; display: inline-block; text-align: center;"><span style="padding: 0px !important; margin: 0px !important; text-indent: 0px !important; display: inline-block !important; vertical-align: baseline !important; font-size: 1px !important;"><span id="li_ui_li_gen_1507763900895_2"><a id="li_ui_li_gen_1507763900895_2-link" href="javascript:void(0);"><span id="li_ui_li_gen_1507763900895_2-logo">in</span><span id="li_ui_li_gen_1507763900895_2-title"><span id="li_ui_li_gen_1507763900895_2-mark"></span><span id="li_ui_li_gen_1507763900895_2-title-text">Share</span></span></a></span></span><span style="padding: 0px !important; margin: 0px !important; text-indent: 0px !important; display: inline-block !important; vertical-align: baseline !important; font-size: 1px !important;"><span id="li_ui_li_gen_1507763900898_3-container" class="IN-right"><span id="li_ui_li_gen_1507763900898_3" class="IN-right"><span id="li_ui_li_gen_1507763900898_3-inner" class="IN-right"><span id="li_ui_li_gen_1507763900898_3-content" class="IN-right">10</span></span></span></span></span></span><script type="in/share+init" data-url="https://www.r-bloggers.com/genome-wide-association-studies-in-r/" data-counter="right"></script></div></div><div style="clear:both"></div></div></div>

	</div><!-- #post-## -->



   				<div id="comments">


<div id="comment-user-details">





		<p class="nocomments">Comments are closed.</p>



</div>

</div><!-- #comments -->

       
	</div>
    <!-- begin second sidebar -->
    <div id="secondsidebar">

		
		<div class="side-widget"><h2>Search R-bloggers</h2><div class="wgs_wrapper"><div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 " style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; outline: none; background: url(&quot;https://www.google.com/cse/static/images/1x/googlelogo_lightgrey_46x16dp.png&quot;) left center no-repeat rgb(255, 255, 255); text-indent: 48px;" x-webkit-speech="" x-webkit-grammar="builtin:search" lang="en" dir="ltr" spellcheck="false" placeholder="Custom Search"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;"><span class="gscb_a" id="gs_cb50">×</span></a></div></td></tr></tbody></table></div><input type="hidden" name="bgresponse" id="bgresponse"></td><td class="gsc-search-button"><input type="image" src="./Genome-wide association studies in R _ R-bloggers_files/search_box_icon.png" class="gsc-search-button gsc-search-button-v2" title="search"></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table><table cellspacing="0" cellpadding="0" class="gsc-branding"><tbody><tr><td class="gsc-branding-user-defined"></td><td class="gsc-branding-text"><div class="gsc-branding-text">powered by</div></td><td class="gsc-branding-img"><img src="./Genome-wide association studies in R _ R-bloggers_files/googlelogo_grey_46x15dp.png" class="gsc-branding-img" srcset="https://www.google.com/cse/static/images/2x/googlelogo_grey_46x15dp.png 2x" scale="0"></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-tabsAreaInvisible"><div class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Web</div><span class="gs-spacer"> </span><div tabindex="0" class=" gsc-tabHeader gsc-tabhInactive gsc-inline-block">Image</div><span class="gs-spacer"> </span></div><div class="gsc-refinementsAreaInvisible"></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><table cellspacing="0" cellpadding="0" class="gsc-resultsHeader"><tbody><tr><td class="gsc-twiddleRegionCell"><div class="gsc-twiddle"><div class="gsc-title">Web</div></div><div class="gsc-stats"></div><div class="gsc-results-selector gsc-all-results-active"><div class="gsc-result-selector gsc-one-result" title="show one result">&nbsp;</div><div class="gsc-result-selector gsc-more-results" title="show more results">&nbsp;</div><div class="gsc-result-selector gsc-all-results" title="show all results">&nbsp;</div></div></td><td class="gsc-configLabelCell"></td></tr></tbody></table><div><div class="gsc-expansionArea"></div></div></div><div class="gsc-resultsRoot gsc-tabData gsc-tabdInactive"><table cellspacing="0" cellpadding="0" class="gsc-resultsHeader"><tbody><tr><td class="gsc-twiddleRegionCell"><div class="gsc-twiddle"><div class="gsc-title">Image</div></div><div class="gsc-stats"></div><div class="gsc-results-selector gsc-all-results-active"><div class="gsc-result-selector gsc-one-result" title="show one result">&nbsp;</div><div class="gsc-result-selector gsc-more-results" title="show more results">&nbsp;</div><div class="gsc-result-selector gsc-all-results" title="show all results">&nbsp;</div></div></td><td class="gsc-configLabelCell"></td></tr></tbody></table><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div><div class="side-widget"><h2>Recent popular posts</h2><ul>				<li>
										<a href="https://www.r-bloggers.com/rstudio-v1-1-released/" class="bump-view" data-bump-view="tp">
						RStudio v1.1 Released					</a>
									</li>
							<li>
										<a href="https://www.r-bloggers.com/data-visualization-course-for-first-year-students/" class="bump-view" data-bump-view="tp">
						Data Visualization Course for First-Year Students					</a>
									</li>
							<li>
										<a href="https://www.r-bloggers.com/linear-logistic-regression-in-r-dealing-with-unknown-factor-levels-in-test-data/" class="bump-view" data-bump-view="tp">
						Linear / Logistic Regression in R: Dealing With Unknown Factor Levels in Test Data					</a>
									</li>
			</ul></div><div class="side-widget"><h2>Most visited articles of the week</h2>
<ol class="wppp_list">
	<li><a href="https://www.r-bloggers.com/how-to-write-the-first-for-loop-in-r/" title="How to write the first for loop in R">How to write the first for loop in R</a></li>
	<li><a href="https://www.r-bloggers.com/installing-r-packages/" title="Installing R packages">Installing R packages</a></li>
	<li><a href="https://www.r-bloggers.com/create-powerpoint-presentations-from-r-with-the-officer-package/" title="Create Powerpoint presentations from R with the OfficeR package">Create Powerpoint presentations from R with the OfficeR package</a></li>
	<li><a href="https://www.r-bloggers.com/how-to-make-a-histogram-with-basic-r/" title="How to Make a Histogram with Basic R">How to Make a Histogram with Basic R</a></li>
	<li><a href="https://www.r-bloggers.com/using-apply-sapply-lapply-in-r/" title="Using apply, sapply, lapply in R">Using apply, sapply, lapply in R</a></li>
	<li><a href="https://www.r-bloggers.com/how-to-perform-a-logistic-regression-in-r/" title="How to perform a Logistic Regression in R">How to perform a Logistic Regression in R</a></li>
	<li><a href="https://www.r-bloggers.com/writing-academic-articles-using-r-markdown-and-latex/" title="Writing academic articles using R Markdown and LaTeX">Writing academic articles using R Markdown and LaTeX</a></li>
	<li><a href="https://www.r-bloggers.com/how-to-learn-r-2/" title="Tutorials for learning R">Tutorials for learning R</a></li>
	<li><a href="https://www.r-bloggers.com/dashboard-design-8-types-of-online-dashboards/" title="Dashboard Design: 8 Types of Online Dashboards">Dashboard Design: 8 Types of Online Dashboards</a></li>
</ol>
</div><div class="side-widget"><h2>Sponsors</h2>			<div class="textwidget"><script data-cfasync="false" type="text/javascript">
// https://support.cloudflare.com/hc/en-us/articles/200169436-How-can-I-have-Rocket-Loader-ignore-my-script-s-in-Automatic-Mode-
// this must be placed higher. Otherwise it doesn't work.
// data-cfasync="false" is for making sure cloudflares' rocketcache doesn't interfeare with this
// in this case it only works because it was used at the original script in the text widget


function createCookie(name,value,days) {
    var expires = "";
    if (days) {
        var date = new Date();
        date.setTime(date.getTime() + (days*24*60*60*1000));
        expires = "; expires=" + date.toUTCString();
    }
    document.cookie = name + "=" + value + expires + "; path=/";
}

function readCookie(name) {
    var nameEQ = name + "=";
    var ca = document.cookie.split(';');
    for(var i=0;i < ca.length;i++) {
        var c = ca[i];
        while (c.charAt(0)==' ') c = c.substring(1,c.length);
        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
    }
    return null;
}

function eraseCookie(name) {
    createCookie(name,"",-1);
}


function readTextFile(file)
{
	// Helps people browse between pages without the need to keep downloading the same 
	// ads txt page everytime. This way, it allows them to use their browser's cache.
	var random_number = readCookie("ad_random_number_cookie");
	if(random_number == null) {
		var random_number = Math.floor(Math.random()*100*(new Date().getTime()/1000));
		createCookie("ad_random_number_cookie",random_number,1)
	}
	
    file += '?t='+random_number;
    var rawFile = new XMLHttpRequest();
    rawFile.onreadystatechange = function ()
    {
        if(rawFile.readyState === 4)
        {
            if(rawFile.status === 200 || rawFile.status == 0)
            {
                // var allText = rawFile.responseText;
                // document.write(allText);
                document.write(rawFile.responseText);
            }
        }
    }
    rawFile.open("GET", file, false);
    rawFile.send(null);
}

// readTextFile('https://raw.githubusercontent.com/Raynos/file-store/master/temp.txt');

readTextFile("https://www.r-bloggers.com/wp-content/uploads/text-widget_anti-cache.txt");

</script><a href="https://earlconf.com/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.mango-solutions.com&#39;, &#39;mango-solutions.com&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/EARL-Tickets-Rbloggers-ad.png" alt="Mango solutions" scale="0">
</a>



<br><hr>
<a href="https://www.datacamp.com/courses/free-introduction-to-r/?tap_a=5644-dce66f&amp;tap_s=10907-287229" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.datacamp.com/courses/free-introduction-to-r/?tap_a=5644-dce66f&amp;tap_s=10907-287229&#39;, &#39;datacamp.com&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/ads_dc.001.jpeg" scale="0">
</a>





<br><hr>
<a href="https://plotcon.plot.ly/workshops" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://plotcon.plot.ly/workshops&#39;, &#39;plot.ly&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/2017-09-05-masterclass-r-shiny-dash-plotly-01.png" scale="0">
</a>








<br><hr>

<a href="https://www.rstudio.com/products/connect/?utm_source=rblogger&amp;utm_medium=connectbannertosite&amp;utm_campaign=connect" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.rstudio.com/products/connect/?utm_source=rblogger&amp;utm_medium=connectbannertosite&amp;utm_campaign=connect&#39;, &#39;rstudio.com&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/RStudioConnect.png" alt="" title="RStudio" scale="0">
</a>



<br><hr>
<a href="http://www.highstat.com/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.highstat.com/&#39;, &#39;highstat.com&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/SouthamptBanffV3.jpg" alt="Zero Inflated Models and Generalized Linear Mixed Models with R" scale="0">
</a>





<br><hr>
<a href="http://www.statisticsviews.com/view/index.html?utm_source=RBlogger&amp;utm_medium=banner&amp;utm_term=StatViews&amp;utm_content=StatViewsAd&amp;utm_campaign=webbanner" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.statisticsviews.com/view/index.html?utm_source=RBlogger&amp;utm_medium=banner&amp;utm_term=StatViews&amp;utm_content=StatViewsAd&amp;utm_campaign=webbanner&#39;, &#39;statisticsviews.com&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/291316_300x125_Final-Web-Ready.jpg" scale="0">
</a>




<br><hr>
<a href="http://www.quantide.com/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.quantide.com&#39;, &#39;quantide.com&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/quantide-2016-04-15.png" alt="Quantide: statistical consulting and training" scale="0">
</a>



<br><hr>
<a href="https://hubs.ly/H08L4p90" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.odsc.com/california&#39;, &#39;odsc.com/california&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/RB-West-clouds_300x125.png" alt="ODSC2" scale="0">
</a>



<br><hr>
<a href="https://hubs.ly/H08L4470" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.odsc.com/london&#39;, &#39;odsc.com/london&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/XaI-West.png" alt="ODSC1" scale="0">
</a>








<br><hr>
<a href="http://www.datasociety.co/coursedescriptions/?utm_medium=advert&amp;utm_source=rbloggers&amp;utm_campaign=august3" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.datasociety.co/coursedescriptions/?utm_medium=advert&amp;utm_source=rbloggers&amp;utm_campaign=august3&#39;, &#39;datasociety.co&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/Rbloggers_Data-Society-300-125.png" alt="datasociety" scale="0">
</a>




<br><hr>
<a href="http://www.eoda.de/de/R-Kenntnis-Tage.html" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.eoda.de/de/R-Kenntnis-Tage.html&#39;, &#39;eoda.de/de/R-Kenntnis-Tage.html&#39;]);"> 
<img src="./Genome-wide association studies in R _ R-bloggers_files/R-Kenntnis-Tage_eodaGmbH-1.png" alt="http://www.eoda.de" scale="0">
</a>







<br><hr>
<a href="http://fantasyfootballanalytics.net/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://fantasyfootballanalytics.net/&#39;, &#39;fantasyfootballanalytics.net&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/FFA-Banner.jpg" scale="0">
</a>








<br><hr>
<a href="http://info.nycdatascience.com/become-a-data-scientist-in-12-week-nyc-data-science-bootcamp-r-blogger" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://info.nycdatascience.com/become-a-data-scientist-in-12-week-nyc-data-science-bootcamp-r-blogger&#39;, &#39;nycdatascience.com&#39;]);">
<img src="./Genome-wide association studies in R _ R-bloggers_files/r-blogger-NYCDSA-300-125-2016-05-02.png" alt="" title="" scale="0">
</a>



<br><hr>
<a href="https://www.crcpress.com/go/r_programming_freebook?utm_source=banner_ad&amp;utm_medium=banner&amp;utm_campaign=170200604" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.crcpress.com/go/r_programming_freebook?utm_source=banner_ad&amp;utm_medium=banner&amp;utm_campaign=170200604&#39;, &#39;crcpress.com&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/R-Blogger-Banner-Ad_v2-2017-02-03.png" alt="CRC R books series" title="CRC R books series" scale="0">
</a>





<br><hr>
<a href="http://www.jumpingrivers.com/?utm_source=rbloggers&amp;utm_campaign=1&amp;utm_medium=banner" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.jumpingrivers.com/&#39;, &#39;jumpingrivers.com&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/jr-AD300_125.jpg" alt="" title="STATWORX" scale="0">
</a>




<br><hr>
<a href="http://www.statworx.de/r-statistik/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.statworx.de/r-statistik/&#39;, &#39;statworx.de&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/STATWORX-R-Bloggers.png" alt="" title="STATWORX" scale="0">
</a>


<br><hr>
<a href="http://yottamine.com/try-yottamine-now" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://yottamine.com/try-yottamine-now&#39;, &#39;yottamine.com&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/R-Blog_Speed.png" alt="" title="yottamine" scale="0">
</a>




<br><hr>
<a href="http://www.6sigma.us/six-sigma-online-training.php" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;http://www.6sigma.us/six-sigma-online-training.php&#39;, &#39;6sigma.us&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/sixsigma300x125-1.jpg" alt="Six Sigma Online Training" title="6 sigma" scale="0">
</a>




<br><hr>
<a href="https://www.udemy.com/courses/?pmtag=&amp;utm_source=growth-affiliate&amp;utm_medium=linkshare&amp;utm_term=eO5bByfCCNo&amp;utm_content=&amp;utm_campaign=_._pn_eO5bByfCCNo_._ci__._ex_Y_._&amp;siteID=eO5bByfCCNo-HqFY4iAUIhzjwVKLZLftGA&amp;LSNPUBID=eO5bByfCCNo" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.udemy.com/courses/?pmtag=&amp;utm_source=growth-affiliate&amp;utm_medium=linkshare&amp;utm_term=eO5bByfCCNo&amp;utm_content=&amp;utm_campaign=_._pn_eO5bByfCCNo_._ci__._ex_Y_._&amp;siteID=eO5bByfCCNo-HqFY4iAUIhzjwVKLZLftGA&amp;LSNPUBID=eO5bByfCCNo&#39;, &#39;linksynergy.com - udemy&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/R-Bloggers-300x125-1.png" alt="" title="R in Udemy" scale="0">
</a>



<br><hr>
<a href="https://www.statcon.de/shop/de/kurse/r" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.statcon.de/shop/de/kurse/r&#39;, &#39;statcon.de&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/2017-07-29-statcon.de-rect4634.png" alt="statcon.de" title="statcon.de" scale="0">
</a>



<br><hr>
<a href="https://mljar.com/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://mljar.com/&#39;, &#39;mljar.com&#39;]);">
    <img src="./Genome-wide association studies in R _ R-bloggers_files/mljar_300x125.jpg" alt="mljar.com" title="mljar.com" scale="0">
</a>




<br><br><hr>



<script type="text/javascript" data-cfasync="false">
// https://support.cloudflare.com/hc/en-us/articles/200169436-How-can-I-have-Rocket-Loader-ignore-my-script-s-in-Automatic-Mode-
// this must be placed higher. Otherwise it doesn't work.
// data-cfasync="false" is for making sure cloudflares' rocketcache doesn't interfeare with this
// in this case it only works because it was used at the original script in the text widget

// document.write("<p>text1");
/* VRTZ JS   300 x 250 pagead  DO NOT MODIFY  */
document.write('<s'+'cript src="//banner.vrtzads.com/scripts/pageads.js?vzId=VZJ130060V994954&vzR='+Math.floor(Math.random()*100*(new Date().getTime()/1000))+'"></s'+'cript>');
// document.write("text2</p>");
</script><script src="https://banner.vrtzads.com/scripts/pageads.js?vzId=VZJ130060V994954&amp;vzR=33746879253"></script>



<br><hr>
<p>
<a href="https://www.r-bloggers.com/contact-us/" rel="nofollow" target="_blank" onclick="_gaq.push([&#39;_trackEvent&#39;, &#39;outbound-widget-ads&#39;, &#39;https://www.r-bloggers.com/contact-us/&#39;, &#39;r-bloggers.com/contact-us&#39;]);">Contact us </a> if you wish to help support R-bloggers, and place <strong>your banner here</strong>.
</p>




<!-- Google Analytics -->
<script>
// https://developers.google.com/analytics/devguides/collection/analyticsjs/

(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-419807-53', 'auto');
// ga('send', 'pageview');
</script>
<!-- End Google Analytics -->






</div>
		</div><div class="side-widget"><h2><a class="rsswidget" href="https://feeds.feedburner.com/Rjobs"><img class="rss-widget-icon" style="border:0" width="14" height="14" src="./Genome-wide association studies in R _ R-bloggers_files/rss.png" alt="RSS" scale="0"></a> <a class="rsswidget" href="https://www.r-users.com/">Jobs for R users</a></h2><ul><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/OMCwwBo1veo/">Quantitative Econometrician @ San Francisco, California, U.S.</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/X3DpiECQ8DE/">Postdoctoral Research Fellow in Healthcare Systems Engineering @ Maryland, U.S.</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/xPASaTEn8kE/">R Programmer &amp; Statistician for Academic Research</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/8QL91gt7zzc/">Data Scientist</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/oyehz_z-zrw/">Postdoctoral Data Scientist: GIS and mHealth</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/5q4Hn0zGhbQ/">Data Scientist @ London</a></li><li><a class="rsswidget" href="http://feedproxy.google.com/~r/RJobs/~3/q46dWx5c0lg/">Crafty developer: R-&gt;C++</a></li></ul></div><div class="side-widget">			<div class="textwidget"><strong><a href="https://www.r-bloggers.com/blogs-list/">Full list of contributing R-bloggers</a></strong></div>
		</div>    </div>
<!-- end second sidebar --></div>
<!-- begin footer -->
<div id="footer">
	<strong><a href="https://www.r-bloggers.com/">R-bloggers</a></strong> was founded by <a href="http://www.r-statistics.com/about/">Tal Galili</a>, with gratitude to the <a href="http://www.r-project.org/">R</a> community.  <br>
    Is powered by <a href="http://www.wordpress.org/">WordPress</a> using a <a href="http://themes.bavotasan.com/" rel="nofollow">bavotasan.com</a> design.<br>
    Copyright © 2017 <strong>R-bloggers</strong>. All Rights Reserved. <a href="http://www.r-bloggers.com/terms/">Terms and Conditions</a> for this website<br>
</div>

<!--
TPC! Memory Usage (http://webjawns.com)
Memory Usage: 64998160
Memory Peak Usage: 65868792
WP Memory Limit: 820M
PHP Memory Limit: 800M
Checkpoints: 11
-->

		<div class="wpusb wpusb-buttons wpusb-fixed-right   wpusb-fixed wpusb-layout-buttons-content wpusb-fixed-position_fixed" id="wpusb-container-fixed" data-element-url="https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F%3Futm_source%3Dshare_buttons%26utm_medium%3Dsocial_media%26utm_campaign%3Dsocial_share" data-element-title="Genome-wide%20association%20studies%20in%20R" data-attr-reference="155868" data-is-term="0" data-element="fixed" data-attr-nonce="7970965" data-disabled-share-counts="1" data-wpusb-component="counter-social-share">

			<div data-element="buttons" class="wpusb-fixed-right-container ">
						<div class="wpusb-item wpusb-facebook ">

				<a href="https://www.facebook.com/sharer.php?u=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F%3Futm_source%3Dshare_buttons%26utm_medium%3Dsocial_media%26utm_campaign%3Dsocial_share" target="_blank" data-action="open-popup" class="wpusb-layout-buttons wpusb-button wpusb-btn " title="Share on Facebook" rel="nofollow">

				   <i class="wpusb-icon-facebook-buttons "></i>
				</a>
			</div>			<div class="wpusb-item wpusb-twitter ">

				<a href="https://twitter.com/share?url=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F%3Futm_source%3Dshare_buttons%26utm_medium%3Dsocial_media%26utm_campaign%3Dsocial_share&amp;text=Genome-wide%20association%20studies%20in%20R%20#rstats #datascience&amp;via=rbloggers" target="_blank" data-action="open-popup" class="wpusb-layout-buttons wpusb-button wpusb-btn " title="Tweet" rel="nofollow">

				   <i class="wpusb-icon-twitter-buttons "></i>
				</a>
			</div>			<div class="wpusb-item wpusb-linkedin ">

				<a href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F%3Futm_source%3Dshare_buttons%26utm_medium%3Dsocial_media%26utm_campaign%3Dsocial_share&amp;title=Genome-wide%20association%20studies%20in%20R" target="_blank" data-action="open-popup" class="wpusb-layout-buttons wpusb-button wpusb-btn " title="Share on Linkedin" rel="nofollow">

				   <i class="wpusb-icon-linkedin-buttons "></i>
				</a>
			</div>			<div class="wpusb-item wpusb-google-plus ">

				<a href="https://plus.google.com/share?url=https%3A%2F%2Fwww.r-bloggers.com%2Fgenome-wide-association-studies-in-r%2F%3Futm_source%3Dshare_buttons%26utm_medium%3Dsocial_media%26utm_campaign%3Dsocial_share" target="_blank" data-action="open-popup" class="wpusb-layout-buttons wpusb-button wpusb-btn " title="Share on Google+" rel="nofollow">

				   <i class="wpusb-icon-google-plus-buttons "></i>
				</a>
			</div>				</div>
				<span class="wpusb-toggle" data-action="close-buttons"></span>
			</div>    <script>
        var snp_f = [];
        var snp_hostname = new RegExp(location.host);
        var snp_http = new RegExp("^(http|https)://", "i");
        var snp_cookie_prefix = '';
        var snp_separate_cookies = false;
        var snp_ajax_url = 'https://www.r-bloggers.com/wp-admin/admin-ajax.php';
        var snp_ignore_cookies = false;
        var snp_enable_analytics_events = false;
        var snp_enable_mobile = false;
        var snp_use_in_all = false;
        var snp_excluded_urls = [];
        snp_excluded_urls.push('');    </script>
    <div class="snp-root">
        <input type="hidden" id="snp_popup" value="">
        <input type="hidden" id="snp_popup_id" value="">
        <input type="hidden" id="snp_popup_theme" value="">
        <input type="hidden" id="snp_exithref" value="">
        <input type="hidden" id="snp_exittarget" value="">
    	<div id="snppopup-welcome" class="snp-pop-109583 snppopup"><input type="hidden" class="snp_open" value="scroll"><input type="hidden" class="snp_show_on_exit" value="2"><input type="hidden" class="snp_exit_js_alert_text" value=""><input type="hidden" class="snp_open_scroll" value="50"><input type="hidden" class="snp_optin_redirect_url" value=""><input type="hidden" class="snp_show_cb_button" value="yes"><input type="hidden" class="snp_popup_id" value="109583"><input type="hidden" class="snp_popup_theme" value="theme6"><input type="hidden" class="snp_overlay" value="disabled"><input type="hidden" class="snp_cookie_conversion" value="30"><input type="hidden" class="snp_cookie_close" value="180"><div class="snp-fb snp-theme6">
    <div class="snp-subscribe-inner">
	<h1 class="snp-header"><i>Never miss an update! </i>
<br>
<strong>Subscribe to R-bloggers</strong> to receive <br>e-mails with the latest R posts.<br>

<small>(You will not see this message again.)</small></h1>	<div class="snp-form">
	    <form action="https://feedburner.google.com/fb/a/mailverify?uri=RBloggers" method="post" class="snp-subscribeform snp_subscribeform">
				<fieldset>
		    <div class="snp-field">
			<input type="text" name="email" id="snp_email" placeholder="Your E-mail..." class="snp-field snp-field-email">		
		    </div>
		    <button type="submit" class="snp-submit">Submit</button>
		</fieldset>
	    </form>
	</div>
	<a href="https://www.r-bloggers.com/genome-wide-association-studies-in-r/#" class="snp_nothanks snp-close">Click here to close (This popup will not appear again)</a>    </div>
    </div>
<style>.snp-pop-109583 .snp-theme6 { max-width: 700px;}
.snp-pop-109583 .snp-theme6 h1 {font-size: 17px;}
.snp-pop-109583 .snp-theme6 { color: #a0a4a9;}
.snp-pop-109583 .snp-theme6 .snp-field ::-webkit-input-placeholder { color: #a0a4a9;}
.snp-pop-109583 .snp-theme6 .snp-field :-moz-placeholder { color: #a0a4a9;}
.snp-pop-109583 .snp-theme6 .snp-field :-ms-input-placeholder { color: #a0a4a9;}
.snp-pop-109583  .snp-theme6 .snp-field input { border: 1px solid #a0a4a9;}
.snp-pop-109583 .snp-theme6 .snp-field { color: #000000;}
.snp-pop-109583 .snp-theme6 { background: #f2f2f2;}
</style></div>    </div>
    <script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shCore.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushAS3.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushBash.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushColdFusion.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushClojure.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushCpp.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushCSharp.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushCss.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushDelphi.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushDiff.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushErlang.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushFSharp.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushGroovy.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushJava.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushJavaFX.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushJScript.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushLatex.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushMatlabKey.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushObjC.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushPerl.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushPhp.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushPlain.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushPowerShell.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushPython.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushR.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushRuby.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushScala.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushSql.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushVb.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/shBrushXml.js.download"></script>
<script type="text/javascript">
	(function(){
		var corecss = document.createElement('link');
		var themecss = document.createElement('link');
		var corecssurl = "https://www.r-bloggers.com/wp-content/plugins/syntaxhighlighter/syntaxhighlighter3/styles/shCore.css?ver=3.0.9b";
		if ( corecss.setAttribute ) {
				corecss.setAttribute( "rel", "stylesheet" );
				corecss.setAttribute( "type", "text/css" );
				corecss.setAttribute( "href", corecssurl );
		} else {
				corecss.rel = "stylesheet";
				corecss.href = corecssurl;
		}
		document.getElementsByTagName("head")[0].insertBefore( corecss, document.getElementById("syntaxhighlighteranchor") );
		var themecssurl = "https://www.r-bloggers.com/wp-content/plugins/syntaxhighlighter/syntaxhighlighter3/styles/shThemeDefault.css?ver=3.0.9b";
		if ( themecss.setAttribute ) {
				themecss.setAttribute( "rel", "stylesheet" );
				themecss.setAttribute( "type", "text/css" );
				themecss.setAttribute( "href", themecssurl );
		} else {
				themecss.rel = "stylesheet";
				themecss.href = themecssurl;
		}
		//document.getElementById("syntaxhighlighteranchor").appendChild(themecss);
		document.getElementsByTagName("head")[0].insertBefore( themecss, document.getElementById("syntaxhighlighteranchor") );
	})();
	SyntaxHighlighter.config.strings.expandSource = '+ expand source';
	SyntaxHighlighter.config.strings.help = '?';
	SyntaxHighlighter.config.strings.alert = 'SyntaxHighlighter\n\n';
	SyntaxHighlighter.config.strings.noBrush = 'Can\'t find brush for: ';
	SyntaxHighlighter.config.strings.brushNotHtmlScript = 'Brush wasn\'t configured for html-script option: ';
	SyntaxHighlighter.defaults['pad-line-numbers'] = false;
	SyntaxHighlighter.defaults['toolbar'] = false;
	SyntaxHighlighter.all();
</script>
<link rel="stylesheet" id="wpusb-style-css" href="./Genome-wide association studies in R _ R-bloggers_files/style(1).css" type="text/css" media="all">
<script type="text/javascript">
/* <![CDATA[ */
var scriptParams = {"google_search_engine_id":"005359090438081006639:paz69t-s8ua"};
/* ]]> */
</script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/google_cse_v2.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/jquery.ck.min.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/ninjapopups.min.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/jquery.fancybox.min.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/comment-reply.min.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/effects.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/photon.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/devicepx-jetpack.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/jquery.sonar.min.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/lazy-load.js.download"></script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/wp-embed.min.js.download"></script>
	<div id="fb-root" class=" fb_reset"><div style="position: absolute; top: -10000px; height: 0px; width: 0px;"><div></div></div><div style="position: absolute; top: -10000px; height: 0px; width: 0px;"><div><iframe name="fb_xdm_frame_https" frameborder="0" allowtransparency="true" allowfullscreen="true" scrolling="no" id="fb_xdm_frame_https" aria-hidden="true" title="Facebook Cross Domain Communication Frame" tabindex="-1" src="./Genome-wide association studies in R _ R-bloggers_files/K6RbmRhg2m2.html" style="border: none;"></iframe></div></div></div>
	<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/widgets.js.download"></script><script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/all.js.download"></script><script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/plusone.js.download" gapi_processed="true"></script><script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/in.js.download"></script><script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/e-201741.js.download" async="" defer=""></script>
<script type="text/javascript">
	_stq = window._stq || [];
	_stq.push([ 'view', {v:'ext',j:'1:4.7.1',blog:'11524731',post:'155868',tz:'-6',srv:'www.r-bloggers.com'} ]);
	_stq.push([ 'clickTrackerInit', '11524731', '155868' ]);
</script>
<script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/effects.js(1).download"></script> 
<script type="text/javascript">
/* <![CDATA[ */
jQuery(function(){
	jQuery("ul.sf-menu").supersubs({ 
		minWidth:    12,
		maxWidth:    27,
		extraWidth:  1
	}).superfish({ 
		delay:       100,
		speed:       250 
	});	});
/* ]]> */
</script>





<!-- Dynamic page generated in 0.636 seconds. -->
<!-- Cached page generated by WP-Super-Cache on 2017-10-09 15:23:31 -->

<!-- Compression = gzip --><script type="text/javascript" src="./Genome-wide association studies in R _ R-bloggers_files/MathJax.js(1).download"></script><script id="texAllTheThingsPageScript" type="text/javascript" src="chrome-extension://cbimabofgmfdkicghcadidpemeenbffn/js/pageScript.js" inlinemath="[[&quot;$&quot;,&quot;$&quot;],[&quot;[;&quot;,&quot;;]&quot;]]" displaymath="[[&quot;$$&quot;,&quot;$$&quot;],[&quot;\\[&quot;,&quot;\\]&quot;]]"></script><table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 193px; display: none; top: 288px; position: absolute; left: 719px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table><img src="./Genome-wide association studies in R _ R-bloggers_files/g.gif" alt=":)" width="6" height="5" id="wpstats" scale="0"></body></html>